// vue-tailwind-datepicker/dist/vue-tailwind-datepicker.js
(function(){"use strict";try{if(typeof document<"u"){var r=document.createElement("style");r.appendChild(document.createTextNode('.vtd-datepicker-overlay.open:before{display:block;opacity:.5}.vtd-datepicker:before{--vtd-datepicker: 0px;content:"";position:absolute;top:0;height:1rem;width:1rem;border-width:1px;border-color:#0000001a;--tw-bg-opacity: 1;background-color:rgb(255 255 255 / var(--tw-bg-opacity));--tw-shadow: 0 1px 3px 0 rgb(0 0 0 / .1), 0 1px 2px -1px rgb(0 0 0 / .1);--tw-shadow-colored: 0 1px 3px 0 var(--tw-shadow-color), 0 1px 2px -1px var(--tw-shadow-color);box-shadow:var(--tw-ring-offset-shadow, 0 0 #0000),var(--tw-ring-shadow, 0 0 #0000),var(--tw-shadow)}:is(.dark .vtd-datepicker):before{--tw-border-opacity: 1;border-color:rgb(55 65 81 / var(--tw-border-opacity));--tw-bg-opacity: 1;background-color:rgb(31 41 55 / var(--tw-bg-opacity))}.vtd-datepicker:before{transform:translate(50%,-50%) rotate(-45deg);-webkit-clip-path:polygon(calc(var(--vtd-datepicker) * -1) calc(var(--vtd-datepicker) * -1),calc(100% + var(--vtd-datepicker)) calc(var(--vtd-datepicker) * -1),calc(100% + var(--vtd-datepicker)) calc(100% + var(--vtd-datepicker)));clip-path:polygon(calc(var(--vtd-datepicker) * -1) calc(var(--vtd-datepicker) * -1),calc(100% + var(--vtd-datepicker)) calc(var(--vtd-datepicker) * -1),calc(100% + var(--vtd-datepicker)) calc(100% + var(--vtd-datepicker)))}.vtd-datepicker.place-left:before{left:.25rem}:is(.dark .vtd-datepicker.place-left):before{--tw-border-opacity: 1;border-color:rgb(55 65 81 / var(--tw-border-opacity));--tw-bg-opacity: 1;background-color:rgb(31 41 55 / var(--tw-bg-opacity))}.vtd-datepicker.place-right:before{right:1.25rem}:is(.dark .vtd-datepicker.place-right):before{--tw-border-opacity: 1;border-color:rgb(55 65 81 / var(--tw-border-opacity));--tw-bg-opacity: 1;background-color:rgb(31 41 55 / var(--tw-bg-opacity))}')),document.head.appendChild(r)}}catch(t){console.error("vite-plugin-css-injected-by-js",t)}})();
import { cloneVNode as Zt, h as ve, Fragment as we, inject as De, provide as ne, ref as J, onMounted as tt, watchEffect as _e, computed as le, defineComponent as ue, onUnmounted as Qe, Teleport as qt, reactive as pt, shallowRef as Jt, openBlock as z, createElementBlock as Z, createElementVNode as A, withDirectives as me, vShow as pe, toDisplayString as ae, renderList as We, withModifiers as $e, unref as se, createCommentVNode as ye, createVNode as ee, TransitionGroup as Xt, withCtx as Fe, normalizeClass as Te, Transition as vt, nextTick as Ie, watch as Qt, isProxy as eo, createBlock as He, renderSlot as dt, mergeProps as to, vModelText as oo } from "vue";
function Oe(e, r, ...a) {
  if (e in r) {
    let s = r[e];
    return typeof s == "function" ? s(...a) : s;
  }
  let t = new Error(`Tried to handle "${e}" but there is no handler defined. Only defined handlers are: ${Object.keys(r).map((s) => `"${s}"`).join(", ")}.`);
  throw Error.captureStackTrace && Error.captureStackTrace(t, Oe), t;
}
var Ue = ((e) => (e[e.None = 0] = "None", e[e.RenderStrategy = 1] = "RenderStrategy", e[e.Static = 2] = "Static", e))(Ue || {}), ro = ((e) => (e[e.Unmount = 0] = "Unmount", e[e.Hidden = 1] = "Hidden", e))(ro || {});
function Pe({ visible: e = !0, features: r = 0, ourProps: a, theirProps: t, ...s }) {
  var n;
  let h = yt(t, a), l = Object.assign(s, { props: h });
  if (e || r & 2 && h.static)
    return nt(l);
  if (r & 1) {
    let v = (n = h.unmount) == null || n ? 0 : 1;
    return Oe(v, { 0() {
      return null;
    }, 1() {
      return nt({ ...s, props: { ...h, hidden: !0, style: { display: "none" } } });
    } });
  }
  return nt(l);
}
function nt({ props: e, attrs: r, slots: a, slot: t, name: s }) {
  var n, h;
  let { as: l, ...v } = ao(e, ["unmount", "static"]), m = (n = a.default) == null ? void 0 : n.call(a, t), w = {};
  if (t) {
    let g = !1, O = [];
    for (let [V, B] of Object.entries(t))
      typeof B == "boolean" && (g = !0), B === !0 && O.push(V);
    g && (w["data-headlessui-state"] = O.join(" "));
  }
  if (l === "template") {
    if (m = ht(m ?? []), Object.keys(v).length > 0 || Object.keys(r).length > 0) {
      let [g, ...O] = m ?? [];
      if (!no(g) || O.length > 0)
        throw new Error(['Passing props on "template"!', "", `The current component <${s} /> is rendering a "template".`, "However we need to passthrough the following props:", Object.keys(v).concat(Object.keys(r)).map((x) => x.trim()).filter((x, P, k) => k.indexOf(x) === P).sort((x, P) => x.localeCompare(P)).map((x) => `  - ${x}`).join(`
`), "", "You can apply a few solutions:", ['Add an `as="..."` prop, to ensure that we render an actual element instead of a "template".', "Render a single element as the child so that we can forward the props onto that element."].map((x) => `  - ${x}`).join(`
`)].join(`
`));
      let V = yt((h = g.props) != null ? h : {}, v), B = Zt(g, V);
      for (let x in V)
        x.startsWith("on") && (B.props || (B.props = {}), B.props[x] = V[x]);
      return B;
    }
    return Array.isArray(m) && m.length === 1 ? m[0] : m;
  }
  return ve(l, Object.assign({}, v, w), { default: () => m });
}
function ht(e) {
  return e.flatMap((r) => r.type === we ? ht(r.children) : [r]);
}
function yt(...e) {
  if (e.length === 0)
    return {};
  if (e.length === 1)
    return e[0];
  let r = {}, a = {};
  for (let t of e)
    for (let s in t)
      s.startsWith("on") && typeof t[s] == "function" ? (a[s] != null || (a[s] = []), a[s].push(t[s])) : r[s] = t[s];
  if (r.disabled || r["aria-disabled"])
    return Object.assign(r, Object.fromEntries(Object.keys(a).map((t) => [t, void 0])));
  for (let t in a)
    Object.assign(r, { [t](s, ...n) {
      let h = a[t];
      for (let l of h) {
        if (s instanceof Event && s.defaultPrevented)
          return;
        l(s, ...n);
      }
    } });
  return r;
}
function ao(e, r = []) {
  let a = Object.assign({}, e);
  for (let t of r)
    t in a && delete a[t];
  return a;
}
function no(e) {
  return e == null ? !1 : typeof e.type == "string" || typeof e.type == "object" || typeof e.type == "function";
}
let so = 0;
function lo() {
  return ++so;
}
function Be() {
  return lo();
}
var Ve = ((e) => (e.Space = " ", e.Enter = "Enter", e.Escape = "Escape", e.Backspace = "Backspace", e.Delete = "Delete", e.ArrowLeft = "ArrowLeft", e.ArrowUp = "ArrowUp", e.ArrowRight = "ArrowRight", e.ArrowDown = "ArrowDown", e.Home = "Home", e.End = "End", e.PageUp = "PageUp", e.PageDown = "PageDown", e.Tab = "Tab", e))(Ve || {});
function H(e) {
  var r;
  return e == null || e.value == null ? null : (r = e.value.$el) != null ? r : e.value;
}
let bt = Symbol("Context");
var Le = ((e) => (e[e.Open = 1] = "Open", e[e.Closed = 2] = "Closed", e[e.Closing = 4] = "Closing", e[e.Opening = 8] = "Opening", e))(Le || {});
function gt() {
  return De(bt, null);
}
function uo(e) {
  ne(bt, e);
}
function it(e, r) {
  if (e)
    return e;
  let a = r ?? "button";
  if (typeof a == "string" && a.toLowerCase() === "button")
    return "button";
}
function io(e, r) {
  let a = J(it(e.value.type, e.value.as));
  return tt(() => {
    a.value = it(e.value.type, e.value.as);
  }), _e(() => {
    var t;
    a.value || H(r) && H(r) instanceof HTMLButtonElement && !((t = H(r)) != null && t.hasAttribute("type")) && (a.value = "button");
  }), a;
}
var co = Object.defineProperty, fo = (e, r, a) => r in e ? co(e, r, { enumerable: !0, configurable: !0, writable: !0, value: a }) : e[r] = a, ct = (e, r, a) => (fo(e, typeof r != "symbol" ? r + "" : r, a), a);
class mo {
  constructor() {
    ct(this, "current", this.detect()), ct(this, "currentId", 0);
  }
  set(r) {
    this.current !== r && (this.currentId = 0, this.current = r);
  }
  reset() {
    this.set(this.detect());
  }
  nextId() {
    return ++this.currentId;
  }
  get isServer() {
    return this.current === "server";
  }
  get isClient() {
    return this.current === "client";
  }
  detect() {
    return typeof window > "u" || typeof document > "u" ? "server" : "client";
  }
}
let ot = new mo();
function Ae(e) {
  if (ot.isServer)
    return null;
  if (e instanceof Node)
    return e.ownerDocument;
  if (e != null && e.hasOwnProperty("value")) {
    let r = H(e);
    if (r)
      return r.ownerDocument;
  }
  return document;
}
let st = ["[contentEditable=true]", "[tabindex]", "a[href]", "area[href]", "button:not([disabled])", "iframe", "input:not([disabled])", "select:not([disabled])", "textarea:not([disabled])"].map((e) => `${e}:not([tabindex='-1'])`).join(",");
var Me = ((e) => (e[e.First = 1] = "First", e[e.Previous = 2] = "Previous", e[e.Next = 4] = "Next", e[e.Last = 8] = "Last", e[e.WrapAround = 16] = "WrapAround", e[e.NoScroll = 32] = "NoScroll", e))(Me || {}), et = ((e) => (e[e.Error = 0] = "Error", e[e.Overflow = 1] = "Overflow", e[e.Success = 2] = "Success", e[e.Underflow = 3] = "Underflow", e))(et || {}), po = ((e) => (e[e.Previous = -1] = "Previous", e[e.Next = 1] = "Next", e))(po || {});
function rt(e = document.body) {
  return e == null ? [] : Array.from(e.querySelectorAll(st)).sort((r, a) => Math.sign((r.tabIndex || Number.MAX_SAFE_INTEGER) - (a.tabIndex || Number.MAX_SAFE_INTEGER)));
}
var ut = ((e) => (e[e.Strict = 0] = "Strict", e[e.Loose = 1] = "Loose", e))(ut || {});
function xt(e, r = 0) {
  var a;
  return e === ((a = Ae(e)) == null ? void 0 : a.body) ? !1 : Oe(r, { 0() {
    return e.matches(st);
  }, 1() {
    let t = e;
    for (; t !== null; ) {
      if (t.matches(st))
        return !0;
      t = t.parentElement;
    }
    return !1;
  } });
}
var vo = ((e) => (e[e.Keyboard = 0] = "Keyboard", e[e.Mouse = 1] = "Mouse", e))(vo || {});
typeof window < "u" && typeof document < "u" && (document.addEventListener("keydown", (e) => {
  e.metaKey || e.altKey || e.ctrlKey || (document.documentElement.dataset.headlessuiFocusVisible = "");
}, !0), document.addEventListener("click", (e) => {
  e.detail === 1 ? delete document.documentElement.dataset.headlessuiFocusVisible : e.detail === 0 && (document.documentElement.dataset.headlessuiFocusVisible = "");
}, !0));
let ho = ["textarea", "input"].join(",");
function yo(e) {
  var r, a;
  return (a = (r = e == null ? void 0 : e.matches) == null ? void 0 : r.call(e, ho)) != null ? a : !1;
}
function bo(e, r = (a) => a) {
  return e.slice().sort((a, t) => {
    let s = r(a), n = r(t);
    if (s === null || n === null)
      return 0;
    let h = s.compareDocumentPosition(n);
    return h & Node.DOCUMENT_POSITION_FOLLOWING ? -1 : h & Node.DOCUMENT_POSITION_PRECEDING ? 1 : 0;
  });
}
function Ee(e, r, { sorted: a = !0, relativeTo: t = null, skipElements: s = [] } = {}) {
  var n;
  let h = (n = Array.isArray(e) ? e.length > 0 ? e[0].ownerDocument : document : e == null ? void 0 : e.ownerDocument) != null ? n : document, l = Array.isArray(e) ? a ? bo(e) : e : rt(e);
  s.length > 0 && l.length > 1 && (l = l.filter((B) => !s.includes(B))), t = t ?? h.activeElement;
  let v = (() => {
    if (r & 5)
      return 1;
    if (r & 10)
      return -1;
    throw new Error("Missing Focus.First, Focus.Previous, Focus.Next or Focus.Last");
  })(), m = (() => {
    if (r & 1)
      return 0;
    if (r & 2)
      return Math.max(0, l.indexOf(t)) - 1;
    if (r & 4)
      return Math.max(0, l.indexOf(t)) + 1;
    if (r & 8)
      return l.length - 1;
    throw new Error("Missing Focus.First, Focus.Previous, Focus.Next or Focus.Last");
  })(), w = r & 32 ? { preventScroll: !0 } : {}, g = 0, O = l.length, V;
  do {
    if (g >= O || g + O <= 0)
      return 0;
    let B = m + g;
    if (r & 16)
      B = (B + O) % O;
    else {
      if (B < 0)
        return 3;
      if (B >= O)
        return 1;
    }
    V = l[B], V == null || V.focus(w), g += v;
  } while (V !== h.activeElement);
  return r & 6 && yo(V) && V.select(), 2;
}
function Ke(e, r, a) {
  ot.isServer || _e((t) => {
    document.addEventListener(e, r, a), t(() => document.removeEventListener(e, r, a));
  });
}
function wt(e, r, a) {
  ot.isServer || _e((t) => {
    window.addEventListener(e, r, a), t(() => window.removeEventListener(e, r, a));
  });
}
function go(e, r, a = le(() => !0)) {
  function t(n, h) {
    if (!a.value || n.defaultPrevented)
      return;
    let l = h(n);
    if (l === null || !l.getRootNode().contains(l))
      return;
    let v = function m(w) {
      return typeof w == "function" ? m(w()) : Array.isArray(w) || w instanceof Set ? w : [w];
    }(e);
    for (let m of v) {
      if (m === null)
        continue;
      let w = m instanceof HTMLElement ? m : H(m);
      if (w != null && w.contains(l) || n.composed && n.composedPath().includes(w))
        return;
    }
    return !xt(l, ut.Loose) && l.tabIndex !== -1 && n.preventDefault(), r(n, l);
  }
  let s = J(null);
  Ke("pointerdown", (n) => {
    var h, l;
    a.value && (s.value = ((l = (h = n.composedPath) == null ? void 0 : h.call(n)) == null ? void 0 : l[0]) || n.target);
  }, !0), Ke("mousedown", (n) => {
    var h, l;
    a.value && (s.value = ((l = (h = n.composedPath) == null ? void 0 : h.call(n)) == null ? void 0 : l[0]) || n.target);
  }, !0), Ke("click", (n) => {
    s.value && (t(n, () => s.value), s.value = null);
  }, !0), Ke("touchend", (n) => t(n, () => n.target instanceof HTMLElement ? n.target : null), !0), wt("blur", (n) => t(n, () => window.document.activeElement instanceof HTMLIFrameElement ? window.document.activeElement : null), !0);
}
var Ne = ((e) => (e[e.None = 1] = "None", e[e.Focusable = 2] = "Focusable", e[e.Hidden = 4] = "Hidden", e))(Ne || {});
let ze = ue({ name: "Hidden", props: { as: { type: [Object, String], default: "div" }, features: { type: Number, default: 1 } }, setup(e, { slots: r, attrs: a }) {
  return () => {
    let { features: t, ...s } = e, n = { "aria-hidden": (t & 2) === 2 ? !0 : void 0, style: { position: "fixed", top: 1, left: 1, width: 1, height: 0, padding: 0, margin: -1, overflow: "hidden", clip: "rect(0, 0, 0, 0)", whiteSpace: "nowrap", borderWidth: "0", ...(t & 4) === 4 && (t & 2) !== 2 && { display: "none" } } };
    return Pe({ ourProps: n, theirProps: s, slot: {}, attrs: a, slots: r, name: "Hidden" });
  };
} });
var Se = ((e) => (e[e.Forwards = 0] = "Forwards", e[e.Backwards = 1] = "Backwards", e))(Se || {});
function kt() {
  let e = J(0);
  return wt("keydown", (r) => {
    r.key === "Tab" && (e.value = r.shiftKey ? 1 : 0);
  }), e;
}
function xo(e, r, a, t) {
  ot.isServer || _e((s) => {
    e = e ?? window, e.addEventListener(r, a, t), s(() => e.removeEventListener(r, a, t));
  });
}
let jt = Symbol("ForcePortalRootContext");
function wo() {
  return De(jt, !1);
}
ue({ name: "ForcePortalRoot", props: { as: { type: [Object, String], default: "template" }, force: { type: Boolean, default: !1 } }, setup(e, { slots: r, attrs: a }) {
  return ne(jt, e.force), () => {
    let { force: t, ...s } = e;
    return Pe({ theirProps: s, ourProps: {}, slot: {}, slots: r, attrs: a, name: "ForcePortalRoot" });
  };
} });
function ko(e) {
  let r = Ae(e);
  if (!r) {
    if (e === null)
      return null;
    throw new Error(`[Headless UI]: Cannot find ownerDocument for contextElement: ${e}`);
  }
  let a = r.getElementById("headlessui-portal-root");
  if (a)
    return a;
  let t = r.createElement("div");
  return t.setAttribute("id", "headlessui-portal-root"), r.body.appendChild(t);
}
ue({ name: "Portal", props: { as: { type: [Object, String], default: "div" } }, setup(e, { slots: r, attrs: a }) {
  let t = J(null), s = le(() => Ae(t)), n = wo(), h = De(_t, null), l = J(n === !0 || h == null ? ko(t.value) : h.resolveTarget());
  _e(() => {
    n || h != null && (l.value = h.resolveTarget());
  });
  let v = De(lt, null);
  return tt(() => {
    let m = H(t);
    m && v && Qe(v.register(m));
  }), Qe(() => {
    var m, w;
    let g = (m = s.value) == null ? void 0 : m.getElementById("headlessui-portal-root");
    g && l.value === g && l.value.children.length <= 0 && ((w = l.value.parentElement) == null || w.removeChild(l.value));
  }), () => {
    if (l.value === null)
      return null;
    let m = { ref: t, "data-headlessui-portal": "" };
    return ve(qt, { to: l.value }, Pe({ ourProps: m, theirProps: e, slot: {}, attrs: a, slots: r, name: "Portal" }));
  };
} });
let lt = Symbol("PortalParentContext");
function jo() {
  let e = De(lt, null), r = J([]);
  function a(n) {
    return r.value.push(n), e && e.register(n), () => t(n);
  }
  function t(n) {
    let h = r.value.indexOf(n);
    h !== -1 && r.value.splice(h, 1), e && e.unregister(n);
  }
  let s = { register: a, unregister: t, portals: r };
  return [r, ue({ name: "PortalWrapper", setup(n, { slots: h }) {
    return ne(lt, s), () => {
      var l;
      return (l = h.default) == null ? void 0 : l.call(h);
    };
  } })];
}
let _t = Symbol("PortalGroupContext");
ue({ name: "PortalGroup", props: { as: { type: [Object, String], default: "template" }, target: { type: Object, default: null } }, setup(e, { attrs: r, slots: a }) {
  let t = pt({ resolveTarget() {
    return e.target;
  } });
  return ne(_t, t), () => {
    let { target: s, ...n } = e;
    return Pe({ theirProps: n, ourProps: {}, slot: {}, attrs: r, slots: a, name: "PortalGroup" });
  };
} });
function _o({ defaultContainers: e = [], portals: r, mainTreeNodeRef: a } = {}) {
  let t = J(null), s = Ae(t);
  function n() {
    var h;
    let l = [];
    for (let v of e)
      v !== null && (v instanceof HTMLElement ? l.push(v) : "value" in v && v.value instanceof HTMLElement && l.push(v.value));
    if (r != null && r.value)
      for (let v of r.value)
        l.push(v);
    for (let v of (h = s == null ? void 0 : s.querySelectorAll("html > *, body > *")) != null ? h : [])
      v !== document.body && v !== document.head && v instanceof HTMLElement && v.id !== "headlessui-portal-root" && (v.contains(H(t)) || l.some((m) => v.contains(m)) || l.push(v));
    return l;
  }
  return { resolveContainers: n, contains(h) {
    return n().some((l) => l.contains(h));
  }, mainTreeNodeRef: t, MainTreeNode() {
    return a != null ? null : ve(ze, { features: Ne.Hidden, ref: t });
  } };
}
function $o() {
  let e = J(null);
  return { mainTreeNodeRef: e, MainTreeNode() {
    return ve(ze, { features: Ne.Hidden, ref: e });
  } };
}
var Mo = ((e) => (e[e.Open = 0] = "Open", e[e.Closed = 1] = "Closed", e))(Mo || {});
let $t = Symbol("PopoverContext");
function at(e) {
  let r = De($t, null);
  if (r === null) {
    let a = new Error(`<${e} /> is missing a parent <${Ot.name} /> component.`);
    throw Error.captureStackTrace && Error.captureStackTrace(a, at), a;
  }
  return r;
}
let Mt = Symbol("PopoverGroupContext");
function St() {
  return De(Mt, null);
}
let Dt = Symbol("PopoverPanelContext");
function So() {
  return De(Dt, null);
}
let Ot = ue({ name: "Popover", inheritAttrs: !1, props: { as: { type: [Object, String], default: "div" } }, setup(e, { slots: r, attrs: a, expose: t }) {
  var s;
  let n = J(null);
  t({ el: n, $el: n });
  let h = J(1), l = J(null), v = J(null), m = J(null), w = J(null), g = le(() => Ae(n)), O = le(() => {
    var L, d;
    if (!H(l) || !H(w))
      return !1;
    for (let q of document.querySelectorAll("body > *"))
      if (Number(q == null ? void 0 : q.contains(H(l))) ^ Number(q == null ? void 0 : q.contains(H(w))))
        return !0;
    let o = rt(), b = o.indexOf(H(l)), I = (b + o.length - 1) % o.length, T = (b + 1) % o.length, C = o[I], K = o[T];
    return !((L = H(w)) != null && L.contains(C)) && !((d = H(w)) != null && d.contains(K));
  }), V = { popoverState: h, buttonId: J(null), panelId: J(null), panel: w, button: l, isPortalled: O, beforePanelSentinel: v, afterPanelSentinel: m, togglePopover() {
    h.value = Oe(h.value, { 0: 1, 1: 0 });
  }, closePopover() {
    h.value !== 1 && (h.value = 1);
  }, close(L) {
    V.closePopover();
    let d = (() => L ? L instanceof HTMLElement ? L : L.value instanceof HTMLElement ? H(L) : H(V.button) : H(V.button))();
    d == null || d.focus();
  } };
  ne($t, V), uo(le(() => Oe(h.value, { 0: Le.Open, 1: Le.Closed })));
  let B = { buttonId: V.buttonId, panelId: V.panelId, close() {
    V.closePopover();
  } }, x = St(), P = x == null ? void 0 : x.registerPopover, [k, M] = jo(), y = _o({ mainTreeNodeRef: x == null ? void 0 : x.mainTreeNodeRef, portals: k, defaultContainers: [l, w] });
  function N() {
    var L, d, o, b;
    return (b = x == null ? void 0 : x.isFocusWithinPopoverGroup()) != null ? b : ((L = g.value) == null ? void 0 : L.activeElement) && (((d = H(l)) == null ? void 0 : d.contains(g.value.activeElement)) || ((o = H(w)) == null ? void 0 : o.contains(g.value.activeElement)));
  }
  return _e(() => P == null ? void 0 : P(B)), xo((s = g.value) == null ? void 0 : s.defaultView, "focus", (L) => {
    var d, o;
    L.target !== window && L.target instanceof HTMLElement && h.value === 0 && (N() || l && w && (y.contains(L.target) || (d = H(V.beforePanelSentinel)) != null && d.contains(L.target) || (o = H(V.afterPanelSentinel)) != null && o.contains(L.target) || V.closePopover()));
  }, !0), go(y.resolveContainers, (L, d) => {
    var o;
    V.closePopover(), xt(d, ut.Loose) || (L.preventDefault(), (o = H(l)) == null || o.focus());
  }, le(() => h.value === 0)), () => {
    let L = { open: h.value === 0, close: V.close };
    return ve(we, [ve(M, {}, () => Pe({ theirProps: { ...e, ...a }, ourProps: { ref: n }, slot: L, slots: r, attrs: a, name: "Popover" })), ve(y.MainTreeNode)]);
  };
} }), Do = ue({ name: "PopoverButton", props: { as: { type: [Object, String], default: "button" }, disabled: { type: [Boolean], default: !1 }, id: { type: String, default: () => `headlessui-popover-button-${Be()}` } }, inheritAttrs: !1, setup(e, { attrs: r, slots: a, expose: t }) {
  let s = at("PopoverButton"), n = le(() => Ae(s.button));
  t({ el: s.button, $el: s.button }), tt(() => {
    s.buttonId.value = e.id;
  }), Qe(() => {
    s.buttonId.value = null;
  });
  let h = St(), l = h == null ? void 0 : h.closeOthers, v = So(), m = le(() => v === null ? !1 : v.value === s.panelId.value), w = J(null), g = `headlessui-focus-sentinel-${Be()}`;
  m.value || _e(() => {
    s.button.value = w.value;
  });
  let O = io(le(() => ({ as: e.as, type: r.type })), w);
  function V(y) {
    var N, L, d, o, b;
    if (m.value) {
      if (s.popoverState.value === 1)
        return;
      switch (y.key) {
        case Ve.Space:
        case Ve.Enter:
          y.preventDefault(), (L = (N = y.target).click) == null || L.call(N), s.closePopover(), (d = H(s.button)) == null || d.focus();
          break;
      }
    } else
      switch (y.key) {
        case Ve.Space:
        case Ve.Enter:
          y.preventDefault(), y.stopPropagation(), s.popoverState.value === 1 && (l == null || l(s.buttonId.value)), s.togglePopover();
          break;
        case Ve.Escape:
          if (s.popoverState.value !== 0)
            return l == null ? void 0 : l(s.buttonId.value);
          if (!H(s.button) || (o = n.value) != null && o.activeElement && !((b = H(s.button)) != null && b.contains(n.value.activeElement)))
            return;
          y.preventDefault(), y.stopPropagation(), s.closePopover();
          break;
      }
  }
  function B(y) {
    m.value || y.key === Ve.Space && y.preventDefault();
  }
  function x(y) {
    var N, L;
    e.disabled || (m.value ? (s.closePopover(), (N = H(s.button)) == null || N.focus()) : (y.preventDefault(), y.stopPropagation(), s.popoverState.value === 1 && (l == null || l(s.buttonId.value)), s.togglePopover(), (L = H(s.button)) == null || L.focus()));
  }
  function P(y) {
    y.preventDefault(), y.stopPropagation();
  }
  let k = kt();
  function M() {
    let y = H(s.panel);
    if (!y)
      return;
    function N() {
      Oe(k.value, { [Se.Forwards]: () => Ee(y, Me.First), [Se.Backwards]: () => Ee(y, Me.Last) }) === et.Error && Ee(rt().filter((L) => L.dataset.headlessuiFocusGuard !== "true"), Oe(k.value, { [Se.Forwards]: Me.Next, [Se.Backwards]: Me.Previous }), { relativeTo: H(s.button) });
    }
    N();
  }
  return () => {
    let y = s.popoverState.value === 0, N = { open: y }, { id: L, ...d } = e, o = m.value ? { ref: w, type: O.value, onKeydown: V, onClick: x } : { ref: w, id: L, type: O.value, "aria-expanded": s.popoverState.value === 0, "aria-controls": H(s.panel) ? s.panelId.value : void 0, disabled: e.disabled ? !0 : void 0, onKeydown: V, onKeyup: B, onClick: x, onMousedown: P };
    return ve(we, [Pe({ ourProps: o, theirProps: { ...r, ...d }, slot: N, attrs: r, slots: a, name: "PopoverButton" }), y && !m.value && s.isPortalled.value && ve(ze, { id: g, features: Ne.Focusable, "data-headlessui-focus-guard": !0, as: "button", type: "button", onFocus: M })]);
  };
} }), Oo = ue({ name: "PopoverOverlay", props: { as: { type: [Object, String], default: "div" }, static: { type: Boolean, default: !1 }, unmount: { type: Boolean, default: !0 } }, setup(e, { attrs: r, slots: a }) {
  let t = at("PopoverOverlay"), s = `headlessui-popover-overlay-${Be()}`, n = gt(), h = le(() => n !== null ? (n.value & Le.Open) === Le.Open : t.popoverState.value === 0);
  function l() {
    t.closePopover();
  }
  return () => {
    let v = { open: t.popoverState.value === 0 };
    return Pe({ ourProps: { id: s, "aria-hidden": !0, onClick: l }, theirProps: e, slot: v, attrs: r, slots: a, features: Ue.RenderStrategy | Ue.Static, visible: h.value, name: "PopoverOverlay" });
  };
} }), Po = ue({ name: "PopoverPanel", props: { as: { type: [Object, String], default: "div" }, static: { type: Boolean, default: !1 }, unmount: { type: Boolean, default: !0 }, focus: { type: Boolean, default: !1 }, id: { type: String, default: () => `headlessui-popover-panel-${Be()}` } }, inheritAttrs: !1, setup(e, { attrs: r, slots: a, expose: t }) {
  let { focus: s } = e, n = at("PopoverPanel"), h = le(() => Ae(n.panel)), l = `headlessui-focus-sentinel-before-${Be()}`, v = `headlessui-focus-sentinel-after-${Be()}`;
  t({ el: n.panel, $el: n.panel }), tt(() => {
    n.panelId.value = e.id;
  }), Qe(() => {
    n.panelId.value = null;
  }), ne(Dt, n.panelId), _e(() => {
    var P, k;
    if (!s || n.popoverState.value !== 0 || !n.panel)
      return;
    let M = (P = h.value) == null ? void 0 : P.activeElement;
    (k = H(n.panel)) != null && k.contains(M) || Ee(H(n.panel), Me.First);
  });
  let m = gt(), w = le(() => m !== null ? (m.value & Le.Open) === Le.Open : n.popoverState.value === 0);
  function g(P) {
    var k, M;
    switch (P.key) {
      case Ve.Escape:
        if (n.popoverState.value !== 0 || !H(n.panel) || h.value && !((k = H(n.panel)) != null && k.contains(h.value.activeElement)))
          return;
        P.preventDefault(), P.stopPropagation(), n.closePopover(), (M = H(n.button)) == null || M.focus();
        break;
    }
  }
  function O(P) {
    var k, M, y, N, L;
    let d = P.relatedTarget;
    d && H(n.panel) && ((k = H(n.panel)) != null && k.contains(d) || (n.closePopover(), ((y = (M = H(n.beforePanelSentinel)) == null ? void 0 : M.contains) != null && y.call(M, d) || (L = (N = H(n.afterPanelSentinel)) == null ? void 0 : N.contains) != null && L.call(N, d)) && d.focus({ preventScroll: !0 })));
  }
  let V = kt();
  function B() {
    let P = H(n.panel);
    if (!P)
      return;
    function k() {
      Oe(V.value, { [Se.Forwards]: () => {
        var M;
        Ee(P, Me.First) === et.Error && ((M = H(n.afterPanelSentinel)) == null || M.focus());
      }, [Se.Backwards]: () => {
        var M;
        (M = H(n.button)) == null || M.focus({ preventScroll: !0 });
      } });
    }
    k();
  }
  function x() {
    let P = H(n.panel);
    if (!P)
      return;
    function k() {
      Oe(V.value, { [Se.Forwards]: () => {
        let M = H(n.button), y = H(n.panel);
        if (!M)
          return;
        let N = rt(), L = N.indexOf(M), d = N.slice(0, L + 1), o = [...N.slice(L + 1), ...d];
        for (let b of o.slice())
          if (b.dataset.headlessuiFocusGuard === "true" || y != null && y.contains(b)) {
            let I = o.indexOf(b);
            I !== -1 && o.splice(I, 1);
          }
        Ee(o, Me.First, { sorted: !1 });
      }, [Se.Backwards]: () => {
        var M;
        Ee(P, Me.Previous) === et.Error && ((M = H(n.button)) == null || M.focus());
      } });
    }
    k();
  }
  return () => {
    let P = { open: n.popoverState.value === 0, close: n.close }, { id: k, focus: M, ...y } = e, N = { ref: n.panel, id: k, onKeydown: g, onFocusout: s && n.popoverState.value === 0 ? O : void 0, tabIndex: -1 };
    return Pe({ ourProps: N, theirProps: { ...r, ...y }, attrs: r, slot: P, slots: { ...a, default: (...L) => {
      var d;
      return [ve(we, [w.value && n.isPortalled.value && ve(ze, { id: l, ref: n.beforePanelSentinel, features: Ne.Focusable, "data-headlessui-focus-guard": !0, as: "button", type: "button", onFocus: B }), (d = a.default) == null ? void 0 : d.call(a, ...L), w.value && n.isPortalled.value && ve(ze, { id: v, ref: n.afterPanelSentinel, features: Ne.Focusable, "data-headlessui-focus-guard": !0, as: "button", type: "button", onFocus: x })])];
    } }, features: Ue.RenderStrategy | Ue.Static, visible: w.value, name: "PopoverPanel" });
  };
} });
ue({ name: "PopoverGroup", inheritAttrs: !1, props: { as: { type: [Object, String], default: "div" } }, setup(e, { attrs: r, slots: a, expose: t }) {
  let s = J(null), n = Jt([]), h = le(() => Ae(s)), l = $o();
  t({ el: s, $el: s });
  function v(O) {
    let V = n.value.indexOf(O);
    V !== -1 && n.value.splice(V, 1);
  }
  function m(O) {
    return n.value.push(O), () => {
      v(O);
    };
  }
  function w() {
    var O;
    let V = h.value;
    if (!V)
      return !1;
    let B = V.activeElement;
    return (O = H(s)) != null && O.contains(B) ? !0 : n.value.some((x) => {
      var P, k;
      return ((P = V.getElementById(x.buttonId.value)) == null ? void 0 : P.contains(B)) || ((k = V.getElementById(x.panelId.value)) == null ? void 0 : k.contains(B));
    });
  }
  function g(O) {
    for (let V of n.value)
      V.buttonId.value !== O && V.close();
  }
  return ne(Mt, { registerPopover: m, unregisterPopover: v, isFocusWithinPopoverGroup: w, closeOthers: g, mainTreeNodeRef: l.mainTreeNodeRef }), () => ve(we, [Pe({ ourProps: { ref: s }, theirProps: { ...e, ...r }, slot: {}, attrs: r, slots: a, name: "PopoverGroup" }), ve(l.MainTreeNode)]);
} });
var Ye = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {};
function Ce(e) {
  return e && e.__esModule && Object.prototype.hasOwnProperty.call(e, "default") ? e.default : e;
}
var Pt = { exports: {} };
(function(e, r) {
  (function(a, t) {
    e.exports = t();
  })(Ye, function() {
    var a = 1e3, t = 6e4, s = 36e5, n = "millisecond", h = "second", l = "minute", v = "hour", m = "day", w = "week", g = "month", O = "quarter", V = "year", B = "date", x = "Invalid Date", P = /^(\d{4})[-/]?(\d{1,2})?[-/]?(\d{0,2})[Tt\s]*(\d{1,2})?:?(\d{1,2})?:?(\d{1,2})?[.:]?(\d+)?$/, k = /\[([^\]]+)]|Y{1,4}|M{1,4}|D{1,2}|d{1,4}|H{1,2}|h{1,2}|a|A|m{1,2}|s{1,2}|Z{1,2}|SSS/g, M = { name: "en", weekdays: "Sunday_Monday_Tuesday_Wednesday_Thursday_Friday_Saturday".split("_"), months: "January_February_March_April_May_June_July_August_September_October_November_December".split("_"), ordinal: function(j) {
      var D = ["th", "st", "nd", "rd"], $ = j % 100;
      return "[" + j + (D[($ - 20) % 10] || D[$] || D[0]) + "]";
    } }, y = function(j, D, $) {
      var Y = String(j);
      return !Y || Y.length >= D ? j : "" + Array(D + 1 - Y.length).join($) + j;
    }, N = { s: y, z: function(j) {
      var D = -j.utcOffset(), $ = Math.abs(D), Y = Math.floor($ / 60), S = $ % 60;
      return (D <= 0 ? "+" : "-") + y(Y, 2, "0") + ":" + y(S, 2, "0");
    }, m: function j(D, $) {
      if (D.date() < $.date())
        return -j($, D);
      var Y = 12 * ($.year() - D.year()) + ($.month() - D.month()), S = D.clone().add(Y, g), E = $ - S < 0, R = D.clone().add(Y + (E ? -1 : 1), g);
      return +(-(Y + ($ - S) / (E ? S - R : R - S)) || 0);
    }, a: function(j) {
      return j < 0 ? Math.ceil(j) || 0 : Math.floor(j);
    }, p: function(j) {
      return { M: g, y: V, w, d: m, D: B, h: v, m: l, s: h, ms: n, Q: O }[j] || String(j || "").toLowerCase().replace(/s$/, "");
    }, u: function(j) {
      return j === void 0;
    } }, L = "en", d = {};
    d[L] = M;
    var o = "$isDayjsObject", b = function(j) {
      return j instanceof K || !(!j || !j[o]);
    }, I = function j(D, $, Y) {
      var S;
      if (!D)
        return L;
      if (typeof D == "string") {
        var E = D.toLowerCase();
        d[E] && (S = E), $ && (d[E] = $, S = E);
        var R = D.split("-");
        if (!S && R.length > 1)
          return j(R[0]);
      } else {
        var G = D.name;
        d[G] = D, S = G;
      }
      return !Y && S && (L = S), S || !Y && L;
    }, T = function(j, D) {
      if (b(j))
        return j.clone();
      var $ = typeof D == "object" ? D : {};
      return $.date = j, $.args = arguments, new K($);
    }, C = N;
    C.l = I, C.i = b, C.w = function(j, D) {
      return T(j, { locale: D.$L, utc: D.$u, x: D.$x, $offset: D.$offset });
    };
    var K = function() {
      function j($) {
        this.$L = I($.locale, null, !0), this.parse($), this.$x = this.$x || $.x || {}, this[o] = !0;
      }
      var D = j.prototype;
      return D.parse = function($) {
        this.$d = function(Y) {
          var S = Y.date, E = Y.utc;
          if (S === null)
            return /* @__PURE__ */ new Date(NaN);
          if (C.u(S))
            return /* @__PURE__ */ new Date();
          if (S instanceof Date)
            return new Date(S);
          if (typeof S == "string" && !/Z$/i.test(S)) {
            var R = S.match(P);
            if (R) {
              var G = R[2] - 1 || 0, Q = (R[7] || "0").substring(0, 3);
              return E ? new Date(Date.UTC(R[1], G, R[3] || 1, R[4] || 0, R[5] || 0, R[6] || 0, Q)) : new Date(R[1], G, R[3] || 1, R[4] || 0, R[5] || 0, R[6] || 0, Q);
            }
          }
          return new Date(S);
        }($), this.init();
      }, D.init = function() {
        var $ = this.$d;
        this.$y = $.getFullYear(), this.$M = $.getMonth(), this.$D = $.getDate(), this.$W = $.getDay(), this.$H = $.getHours(), this.$m = $.getMinutes(), this.$s = $.getSeconds(), this.$ms = $.getMilliseconds();
      }, D.$utils = function() {
        return C;
      }, D.isValid = function() {
        return this.$d.toString() !== x;
      }, D.isSame = function($, Y) {
        var S = T($);
        return this.startOf(Y) <= S && S <= this.endOf(Y);
      }, D.isAfter = function($, Y) {
        return T($) < this.startOf(Y);
      }, D.isBefore = function($, Y) {
        return this.endOf(Y) < T($);
      }, D.$g = function($, Y, S) {
        return C.u($) ? this[Y] : this.set(S, $);
      }, D.unix = function() {
        return Math.floor(this.valueOf() / 1e3);
      }, D.valueOf = function() {
        return this.$d.getTime();
      }, D.startOf = function($, Y) {
        var S = this, E = !!C.u(Y) || Y, R = C.p($), G = function(he, re) {
          var fe = C.w(S.$u ? Date.UTC(S.$y, re, he) : new Date(S.$y, re, he), S);
          return E ? fe : fe.endOf(m);
        }, Q = function(he, re) {
          return C.w(S.toDate()[he].apply(S.toDate("s"), (E ? [0, 0, 0, 0] : [23, 59, 59, 999]).slice(re)), S);
        }, te = this.$W, oe = this.$M, de = this.$D, ce = "set" + (this.$u ? "UTC" : "");
        switch (R) {
          case V:
            return E ? G(1, 0) : G(31, 11);
          case g:
            return E ? G(1, oe) : G(0, oe + 1);
          case w:
            var be = this.$locale().weekStart || 0, ge = (te < be ? te + 7 : te) - be;
            return G(E ? de - ge : de + (6 - ge), oe);
          case m:
          case B:
            return Q(ce + "Hours", 0);
          case v:
            return Q(ce + "Minutes", 1);
          case l:
            return Q(ce + "Seconds", 2);
          case h:
            return Q(ce + "Milliseconds", 3);
          default:
            return this.clone();
        }
      }, D.endOf = function($) {
        return this.startOf($, !1);
      }, D.$set = function($, Y) {
        var S, E = C.p($), R = "set" + (this.$u ? "UTC" : ""), G = (S = {}, S[m] = R + "Date", S[B] = R + "Date", S[g] = R + "Month", S[V] = R + "FullYear", S[v] = R + "Hours", S[l] = R + "Minutes", S[h] = R + "Seconds", S[n] = R + "Milliseconds", S)[E], Q = E === m ? this.$D + (Y - this.$W) : Y;
        if (E === g || E === V) {
          var te = this.clone().set(B, 1);
          te.$d[G](Q), te.init(), this.$d = te.set(B, Math.min(this.$D, te.daysInMonth())).$d;
        } else
          G && this.$d[G](Q);
        return this.init(), this;
      }, D.set = function($, Y) {
        return this.clone().$set($, Y);
      }, D.get = function($) {
        return this[C.p($)]();
      }, D.add = function($, Y) {
        var S, E = this;
        $ = Number($);
        var R = C.p(Y), G = function(oe) {
          var de = T(E);
          return C.w(de.date(de.date() + Math.round(oe * $)), E);
        };
        if (R === g)
          return this.set(g, this.$M + $);
        if (R === V)
          return this.set(V, this.$y + $);
        if (R === m)
          return G(1);
        if (R === w)
          return G(7);
        var Q = (S = {}, S[l] = t, S[v] = s, S[h] = a, S)[R] || 1, te = this.$d.getTime() + $ * Q;
        return C.w(te, this);
      }, D.subtract = function($, Y) {
        return this.add(-1 * $, Y);
      }, D.format = function($) {
        var Y = this, S = this.$locale();
        if (!this.isValid())
          return S.invalidDate || x;
        var E = $ || "YYYY-MM-DDTHH:mm:ssZ", R = C.z(this), G = this.$H, Q = this.$m, te = this.$M, oe = S.weekdays, de = S.months, ce = S.meridiem, be = function(re, fe, xe, ke) {
          return re && (re[fe] || re(Y, E)) || xe[fe].slice(0, ke);
        }, ge = function(re) {
          return C.s(G % 12 || 12, re, "0");
        }, he = ce || function(re, fe, xe) {
          var ke = re < 12 ? "AM" : "PM";
          return xe ? ke.toLowerCase() : ke;
        };
        return E.replace(k, function(re, fe) {
          return fe || function(xe) {
            switch (xe) {
              case "YY":
                return String(Y.$y).slice(-2);
              case "YYYY":
                return C.s(Y.$y, 4, "0");
              case "M":
                return te + 1;
              case "MM":
                return C.s(te + 1, 2, "0");
              case "MMM":
                return be(S.monthsShort, te, de, 3);
              case "MMMM":
                return be(de, te);
              case "D":
                return Y.$D;
              case "DD":
                return C.s(Y.$D, 2, "0");
              case "d":
                return String(Y.$W);
              case "dd":
                return be(S.weekdaysMin, Y.$W, oe, 2);
              case "ddd":
                return be(S.weekdaysShort, Y.$W, oe, 3);
              case "dddd":
                return oe[Y.$W];
              case "H":
                return String(G);
              case "HH":
                return C.s(G, 2, "0");
              case "h":
                return ge(1);
              case "hh":
                return ge(2);
              case "a":
                return he(G, Q, !0);
              case "A":
                return he(G, Q, !1);
              case "m":
                return String(Q);
              case "mm":
                return C.s(Q, 2, "0");
              case "s":
                return String(Y.$s);
              case "ss":
                return C.s(Y.$s, 2, "0");
              case "SSS":
                return C.s(Y.$ms, 3, "0");
              case "Z":
                return R;
            }
            return null;
          }(re) || R.replace(":", "");
        });
      }, D.utcOffset = function() {
        return 15 * -Math.round(this.$d.getTimezoneOffset() / 15);
      }, D.diff = function($, Y, S) {
        var E, R = this, G = C.p(Y), Q = T($), te = (Q.utcOffset() - this.utcOffset()) * t, oe = this - Q, de = function() {
          return C.m(R, Q);
        };
        switch (G) {
          case V:
            E = de() / 12;
            break;
          case g:
            E = de();
            break;
          case O:
            E = de() / 3;
            break;
          case w:
            E = (oe - te) / 6048e5;
            break;
          case m:
            E = (oe - te) / 864e5;
            break;
          case v:
            E = oe / s;
            break;
          case l:
            E = oe / t;
            break;
          case h:
            E = oe / a;
            break;
          default:
            E = oe;
        }
        return S ? E : C.a(E);
      }, D.daysInMonth = function() {
        return this.endOf(g).$D;
      }, D.$locale = function() {
        return d[this.$L];
      }, D.locale = function($, Y) {
        if (!$)
          return this.$L;
        var S = this.clone(), E = I($, Y, !0);
        return E && (S.$L = E), S;
      }, D.clone = function() {
        return C.w(this.$d, this);
      }, D.toDate = function() {
        return new Date(this.valueOf());
      }, D.toJSON = function() {
        return this.isValid() ? this.toISOString() : null;
      }, D.toISOString = function() {
        return this.$d.toISOString();
      }, D.toString = function() {
        return this.$d.toUTCString();
      }, j;
    }(), q = K.prototype;
    return T.prototype = q, [["$ms", n], ["$s", h], ["$m", l], ["$H", v], ["$W", m], ["$M", g], ["$y", V], ["$D", B]].forEach(function(j) {
      q[j[1]] = function(D) {
        return this.$g(D, j[0], j[1]);
      };
    }), T.extend = function(j, D) {
      return j.$i || (j(D, K, T), j.$i = !0), T;
    }, T.locale = I, T.isDayjs = b, T.unix = function(j) {
      return T(1e3 * j);
    }, T.en = d[L], T.Ls = d, T.p = {}, T;
  });
})(Pt);
var Vo = Pt.exports;
const i = /* @__PURE__ */ Ce(Vo);
var Vt = { exports: {} };
(function(e, r) {
  (function(a, t) {
    e.exports = t();
  })(Ye, function() {
    return function(a, t, s) {
      var n = t.prototype, h = function(g) {
        return g && (g.indexOf ? g : g.s);
      }, l = function(g, O, V, B, x) {
        var P = g.name ? g : g.$locale(), k = h(P[O]), M = h(P[V]), y = k || M.map(function(L) {
          return L.slice(0, B);
        });
        if (!x)
          return y;
        var N = P.weekStart;
        return y.map(function(L, d) {
          return y[(d + (N || 0)) % 7];
        });
      }, v = function() {
        return s.Ls[s.locale()];
      }, m = function(g, O) {
        return g.formats[O] || function(V) {
          return V.replace(/(\[[^\]]+])|(MMMM|MM|DD|dddd)/g, function(B, x, P) {
            return x || P.slice(1);
          });
        }(g.formats[O.toUpperCase()]);
      }, w = function() {
        var g = this;
        return { months: function(O) {
          return O ? O.format("MMMM") : l(g, "months");
        }, monthsShort: function(O) {
          return O ? O.format("MMM") : l(g, "monthsShort", "months", 3);
        }, firstDayOfWeek: function() {
          return g.$locale().weekStart || 0;
        }, weekdays: function(O) {
          return O ? O.format("dddd") : l(g, "weekdays");
        }, weekdaysMin: function(O) {
          return O ? O.format("dd") : l(g, "weekdaysMin", "weekdays", 2);
        }, weekdaysShort: function(O) {
          return O ? O.format("ddd") : l(g, "weekdaysShort", "weekdays", 3);
        }, longDateFormat: function(O) {
          return m(g.$locale(), O);
        }, meridiem: this.$locale().meridiem, ordinal: this.$locale().ordinal };
      };
      n.localeData = function() {
        return w.bind(this)();
      }, s.localeData = function() {
        var g = v();
        return { firstDayOfWeek: function() {
          return g.weekStart || 0;
        }, weekdays: function() {
          return s.weekdays();
        }, weekdaysShort: function() {
          return s.weekdaysShort();
        }, weekdaysMin: function() {
          return s.weekdaysMin();
        }, months: function() {
          return s.months();
        }, monthsShort: function() {
          return s.monthsShort();
        }, longDateFormat: function(O) {
          return m(g, O);
        }, meridiem: g.meridiem, ordinal: g.ordinal };
      }, s.months = function() {
        return l(v(), "months");
      }, s.monthsShort = function() {
        return l(v(), "monthsShort", "months", 3);
      }, s.weekdays = function(g) {
        return l(v(), "weekdays", null, null, g);
      }, s.weekdaysShort = function(g) {
        return l(v(), "weekdaysShort", "weekdays", 3, g);
      }, s.weekdaysMin = function(g) {
        return l(v(), "weekdaysMin", "weekdays", 2, g);
      };
    };
  });
})(Vt);
var To = Vt.exports;
const Ao = /* @__PURE__ */ Ce(To);
var Tt = { exports: {} };
(function(e, r) {
  (function(a, t) {
    e.exports = t();
  })(Ye, function() {
    var a = { LTS: "h:mm:ss A", LT: "h:mm A", L: "MM/DD/YYYY", LL: "MMMM D, YYYY", LLL: "MMMM D, YYYY h:mm A", LLLL: "dddd, MMMM D, YYYY h:mm A" };
    return function(t, s, n) {
      var h = s.prototype, l = h.format;
      n.en.formats = a, h.format = function(v) {
        v === void 0 && (v = "YYYY-MM-DDTHH:mm:ssZ");
        var m = this.$locale().formats, w = function(g, O) {
          return g.replace(/(\[[^\]]+])|(LTS?|l{1,4}|L{1,4})/g, function(V, B, x) {
            var P = x && x.toUpperCase();
            return B || O[x] || a[x] || O[P].replace(/(\[[^\]]+])|(MMMM|MM|DD|dddd)/g, function(k, M, y) {
              return M || y.slice(1);
            });
          });
        }(v, m === void 0 ? {} : m);
        return l.call(this, w);
      };
    };
  });
})(Tt);
var Yo = Tt.exports;
const Co = /* @__PURE__ */ Ce(Yo);
var At = { exports: {} };
(function(e, r) {
  (function(a, t) {
    e.exports = t();
  })(Ye, function() {
    var a = { LTS: "h:mm:ss A", LT: "h:mm A", L: "MM/DD/YYYY", LL: "MMMM D, YYYY", LLL: "MMMM D, YYYY h:mm A", LLLL: "dddd, MMMM D, YYYY h:mm A" }, t = /(\[[^[]*\])|([-_:/.,()\s]+)|(A|a|YYYY|YY?|MM?M?M?|Do|DD?|hh?|HH?|mm?|ss?|S{1,3}|z|ZZ?)/g, s = /\d\d/, n = /\d\d?/, h = /\d*[^-_:/,()\s\d]+/, l = {}, v = function(x) {
      return (x = +x) + (x > 68 ? 1900 : 2e3);
    }, m = function(x) {
      return function(P) {
        this[x] = +P;
      };
    }, w = [/[+-]\d\d:?(\d\d)?|Z/, function(x) {
      (this.zone || (this.zone = {})).offset = function(P) {
        if (!P || P === "Z")
          return 0;
        var k = P.match(/([+-]|\d\d)/g), M = 60 * k[1] + (+k[2] || 0);
        return M === 0 ? 0 : k[0] === "+" ? -M : M;
      }(x);
    }], g = function(x) {
      var P = l[x];
      return P && (P.indexOf ? P : P.s.concat(P.f));
    }, O = function(x, P) {
      var k, M = l.meridiem;
      if (M) {
        for (var y = 1; y <= 24; y += 1)
          if (x.indexOf(M(y, 0, P)) > -1) {
            k = y > 12;
            break;
          }
      } else
        k = x === (P ? "pm" : "PM");
      return k;
    }, V = { A: [h, function(x) {
      this.afternoon = O(x, !1);
    }], a: [h, function(x) {
      this.afternoon = O(x, !0);
    }], S: [/\d/, function(x) {
      this.milliseconds = 100 * +x;
    }], SS: [s, function(x) {
      this.milliseconds = 10 * +x;
    }], SSS: [/\d{3}/, function(x) {
      this.milliseconds = +x;
    }], s: [n, m("seconds")], ss: [n, m("seconds")], m: [n, m("minutes")], mm: [n, m("minutes")], H: [n, m("hours")], h: [n, m("hours")], HH: [n, m("hours")], hh: [n, m("hours")], D: [n, m("day")], DD: [s, m("day")], Do: [h, function(x) {
      var P = l.ordinal, k = x.match(/\d+/);
      if (this.day = k[0], P)
        for (var M = 1; M <= 31; M += 1)
          P(M).replace(/\[|\]/g, "") === x && (this.day = M);
    }], M: [n, m("month")], MM: [s, m("month")], MMM: [h, function(x) {
      var P = g("months"), k = (g("monthsShort") || P.map(function(M) {
        return M.slice(0, 3);
      })).indexOf(x) + 1;
      if (k < 1)
        throw new Error();
      this.month = k % 12 || k;
    }], MMMM: [h, function(x) {
      var P = g("months").indexOf(x) + 1;
      if (P < 1)
        throw new Error();
      this.month = P % 12 || P;
    }], Y: [/[+-]?\d+/, m("year")], YY: [s, function(x) {
      this.year = v(x);
    }], YYYY: [/\d{4}/, m("year")], Z: w, ZZ: w };
    function B(x) {
      var P, k;
      P = x, k = l && l.formats;
      for (var M = (x = P.replace(/(\[[^\]]+])|(LTS?|l{1,4}|L{1,4})/g, function(I, T, C) {
        var K = C && C.toUpperCase();
        return T || k[C] || a[C] || k[K].replace(/(\[[^\]]+])|(MMMM|MM|DD|dddd)/g, function(q, j, D) {
          return j || D.slice(1);
        });
      })).match(t), y = M.length, N = 0; N < y; N += 1) {
        var L = M[N], d = V[L], o = d && d[0], b = d && d[1];
        M[N] = b ? { regex: o, parser: b } : L.replace(/^\[|\]$/g, "");
      }
      return function(I) {
        for (var T = {}, C = 0, K = 0; C < y; C += 1) {
          var q = M[C];
          if (typeof q == "string")
            K += q.length;
          else {
            var j = q.regex, D = q.parser, $ = I.slice(K), Y = j.exec($)[0];
            D.call(T, Y), I = I.replace(Y, "");
          }
        }
        return function(S) {
          var E = S.afternoon;
          if (E !== void 0) {
            var R = S.hours;
            E ? R < 12 && (S.hours += 12) : R === 12 && (S.hours = 0), delete S.afternoon;
          }
        }(T), T;
      };
    }
    return function(x, P, k) {
      k.p.customParseFormat = !0, x && x.parseTwoDigitYear && (v = x.parseTwoDigitYear);
      var M = P.prototype, y = M.parse;
      M.parse = function(N) {
        var L = N.date, d = N.utc, o = N.args;
        this.$u = d;
        var b = o[1];
        if (typeof b == "string") {
          var I = o[2] === !0, T = o[3] === !0, C = I || T, K = o[2];
          T && (K = o[2]), l = this.$locale(), !I && K && (l = k.Ls[K]), this.$d = function($, Y, S) {
            try {
              if (["x", "X"].indexOf(Y) > -1)
                return new Date((Y === "X" ? 1e3 : 1) * $);
              var E = B(Y)($), R = E.year, G = E.month, Q = E.day, te = E.hours, oe = E.minutes, de = E.seconds, ce = E.milliseconds, be = E.zone, ge = /* @__PURE__ */ new Date(), he = Q || (R || G ? 1 : ge.getDate()), re = R || ge.getFullYear(), fe = 0;
              R && !G || (fe = G > 0 ? G - 1 : ge.getMonth());
              var xe = te || 0, ke = oe || 0, Re = de || 0, p = ce || 0;
              return be ? new Date(Date.UTC(re, fe, he, xe, ke, Re, p + 60 * be.offset * 1e3)) : S ? new Date(Date.UTC(re, fe, he, xe, ke, Re, p)) : new Date(re, fe, he, xe, ke, Re, p);
            } catch {
              return /* @__PURE__ */ new Date("");
            }
          }(L, b, d), this.init(), K && K !== !0 && (this.$L = this.locale(K).$L), C && L != this.format(b) && (this.$d = /* @__PURE__ */ new Date("")), l = {};
        } else if (b instanceof Array)
          for (var q = b.length, j = 1; j <= q; j += 1) {
            o[1] = b[j - 1];
            var D = k.apply(this, o);
            if (D.isValid()) {
              this.$d = D.$d, this.$L = D.$L, this.init();
              break;
            }
            j === q && (this.$d = /* @__PURE__ */ new Date(""));
          }
        else
          y.call(this, N);
      };
    };
  });
})(At);
var Eo = At.exports;
const Lo = /* @__PURE__ */ Ce(Eo);
var Yt = { exports: {} };
(function(e, r) {
  (function(a, t) {
    e.exports = t();
  })(Ye, function() {
    return function(a, t, s) {
      t.prototype.isToday = function() {
        var n = "YYYY-MM-DD", h = s();
        return this.format(n) === h.format(n);
      };
    };
  });
})(Yt);
var Fo = Yt.exports;
const Bo = /* @__PURE__ */ Ce(Fo);
var Ct = { exports: {} };
(function(e, r) {
  (function(a, t) {
    e.exports = t();
  })(Ye, function() {
    return function(a, t, s) {
      t.prototype.isBetween = function(n, h, l, v) {
        var m = s(n), w = s(h), g = (v = v || "()")[0] === "(", O = v[1] === ")";
        return (g ? this.isAfter(m, l) : !this.isBefore(m, l)) && (O ? this.isBefore(w, l) : !this.isAfter(w, l)) || (g ? this.isBefore(m, l) : !this.isAfter(m, l)) && (O ? this.isAfter(w, l) : !this.isBefore(w, l));
      };
    };
  });
})(Ct);
var No = Ct.exports;
const Ro = /* @__PURE__ */ Ce(No);
var Et = { exports: {} };
(function(e, r) {
  (function(a, t) {
    e.exports = t();
  })(Ye, function() {
    var a, t, s = 1e3, n = 6e4, h = 36e5, l = 864e5, v = /\[([^\]]+)]|Y{1,4}|M{1,4}|D{1,2}|d{1,4}|H{1,2}|h{1,2}|a|A|m{1,2}|s{1,2}|Z{1,2}|SSS/g, m = 31536e6, w = 2628e6, g = /^(-|\+)?P(?:([-+]?[0-9,.]*)Y)?(?:([-+]?[0-9,.]*)M)?(?:([-+]?[0-9,.]*)W)?(?:([-+]?[0-9,.]*)D)?(?:T(?:([-+]?[0-9,.]*)H)?(?:([-+]?[0-9,.]*)M)?(?:([-+]?[0-9,.]*)S)?)?$/, O = { years: m, months: w, days: l, hours: h, minutes: n, seconds: s, milliseconds: 1, weeks: 6048e5 }, V = function(d) {
      return d instanceof N;
    }, B = function(d, o, b) {
      return new N(d, b, o.$l);
    }, x = function(d) {
      return t.p(d) + "s";
    }, P = function(d) {
      return d < 0;
    }, k = function(d) {
      return P(d) ? Math.ceil(d) : Math.floor(d);
    }, M = function(d) {
      return Math.abs(d);
    }, y = function(d, o) {
      return d ? P(d) ? { negative: !0, format: "" + M(d) + o } : { negative: !1, format: "" + d + o } : { negative: !1, format: "" };
    }, N = function() {
      function d(b, I, T) {
        var C = this;
        if (this.$d = {}, this.$l = T, b === void 0 && (this.$ms = 0, this.parseFromMilliseconds()), I)
          return B(b * O[x(I)], this);
        if (typeof b == "number")
          return this.$ms = b, this.parseFromMilliseconds(), this;
        if (typeof b == "object")
          return Object.keys(b).forEach(function(j) {
            C.$d[x(j)] = b[j];
          }), this.calMilliseconds(), this;
        if (typeof b == "string") {
          var K = b.match(g);
          if (K) {
            var q = K.slice(2).map(function(j) {
              return j != null ? Number(j) : 0;
            });
            return this.$d.years = q[0], this.$d.months = q[1], this.$d.weeks = q[2], this.$d.days = q[3], this.$d.hours = q[4], this.$d.minutes = q[5], this.$d.seconds = q[6], this.calMilliseconds(), this;
          }
        }
        return this;
      }
      var o = d.prototype;
      return o.calMilliseconds = function() {
        var b = this;
        this.$ms = Object.keys(this.$d).reduce(function(I, T) {
          return I + (b.$d[T] || 0) * O[T];
        }, 0);
      }, o.parseFromMilliseconds = function() {
        var b = this.$ms;
        this.$d.years = k(b / m), b %= m, this.$d.months = k(b / w), b %= w, this.$d.days = k(b / l), b %= l, this.$d.hours = k(b / h), b %= h, this.$d.minutes = k(b / n), b %= n, this.$d.seconds = k(b / s), b %= s, this.$d.milliseconds = b;
      }, o.toISOString = function() {
        var b = y(this.$d.years, "Y"), I = y(this.$d.months, "M"), T = +this.$d.days || 0;
        this.$d.weeks && (T += 7 * this.$d.weeks);
        var C = y(T, "D"), K = y(this.$d.hours, "H"), q = y(this.$d.minutes, "M"), j = this.$d.seconds || 0;
        this.$d.milliseconds && (j += this.$d.milliseconds / 1e3, j = Math.round(1e3 * j) / 1e3);
        var D = y(j, "S"), $ = b.negative || I.negative || C.negative || K.negative || q.negative || D.negative, Y = K.format || q.format || D.format ? "T" : "", S = ($ ? "-" : "") + "P" + b.format + I.format + C.format + Y + K.format + q.format + D.format;
        return S === "P" || S === "-P" ? "P0D" : S;
      }, o.toJSON = function() {
        return this.toISOString();
      }, o.format = function(b) {
        var I = b || "YYYY-MM-DDTHH:mm:ss", T = { Y: this.$d.years, YY: t.s(this.$d.years, 2, "0"), YYYY: t.s(this.$d.years, 4, "0"), M: this.$d.months, MM: t.s(this.$d.months, 2, "0"), D: this.$d.days, DD: t.s(this.$d.days, 2, "0"), H: this.$d.hours, HH: t.s(this.$d.hours, 2, "0"), m: this.$d.minutes, mm: t.s(this.$d.minutes, 2, "0"), s: this.$d.seconds, ss: t.s(this.$d.seconds, 2, "0"), SSS: t.s(this.$d.milliseconds, 3, "0") };
        return I.replace(v, function(C, K) {
          return K || String(T[C]);
        });
      }, o.as = function(b) {
        return this.$ms / O[x(b)];
      }, o.get = function(b) {
        var I = this.$ms, T = x(b);
        return T === "milliseconds" ? I %= 1e3 : I = T === "weeks" ? k(I / O[T]) : this.$d[T], I || 0;
      }, o.add = function(b, I, T) {
        var C;
        return C = I ? b * O[x(I)] : V(b) ? b.$ms : B(b, this).$ms, B(this.$ms + C * (T ? -1 : 1), this);
      }, o.subtract = function(b, I) {
        return this.add(b, I, !0);
      }, o.locale = function(b) {
        var I = this.clone();
        return I.$l = b, I;
      }, o.clone = function() {
        return B(this.$ms, this);
      }, o.humanize = function(b) {
        return a().add(this.$ms, "ms").locale(this.$l).fromNow(!b);
      }, o.valueOf = function() {
        return this.asMilliseconds();
      }, o.milliseconds = function() {
        return this.get("milliseconds");
      }, o.asMilliseconds = function() {
        return this.as("milliseconds");
      }, o.seconds = function() {
        return this.get("seconds");
      }, o.asSeconds = function() {
        return this.as("seconds");
      }, o.minutes = function() {
        return this.get("minutes");
      }, o.asMinutes = function() {
        return this.as("minutes");
      }, o.hours = function() {
        return this.get("hours");
      }, o.asHours = function() {
        return this.as("hours");
      }, o.days = function() {
        return this.get("days");
      }, o.asDays = function() {
        return this.as("days");
      }, o.weeks = function() {
        return this.get("weeks");
      }, o.asWeeks = function() {
        return this.as("weeks");
      }, o.months = function() {
        return this.get("months");
      }, o.asMonths = function() {
        return this.as("months");
      }, o.years = function() {
        return this.get("years");
      }, o.asYears = function() {
        return this.as("years");
      }, d;
    }(), L = function(d, o, b) {
      return d.add(o.years() * b, "y").add(o.months() * b, "M").add(o.days() * b, "d").add(o.hours() * b, "h").add(o.minutes() * b, "m").add(o.seconds() * b, "s").add(o.milliseconds() * b, "ms");
    };
    return function(d, o, b) {
      a = b, t = b().$utils(), b.duration = function(C, K) {
        var q = b.locale();
        return B(C, { $l: q }, K);
      }, b.isDuration = V;
      var I = o.prototype.add, T = o.prototype.subtract;
      o.prototype.add = function(C, K) {
        return V(C) ? L(this, C, 1) : I.bind(this)(C, K);
      }, o.prototype.subtract = function(C, K) {
        return V(C) ? L(this, C, -1) : T.bind(this)(C, K);
      };
    };
  });
})(Et);
var Io = Et.exports;
const Ho = /* @__PURE__ */ Ce(Io);
var Lt = { exports: {} };
(function(e, r) {
  (function(a, t) {
    e.exports = t();
  })(Ye, function() {
    var a = "week", t = "year";
    return function(s, n, h) {
      var l = n.prototype;
      l.week = function(v) {
        if (v === void 0 && (v = null), v !== null)
          return this.add(7 * (v - this.week()), "day");
        var m = this.$locale().yearStart || 1;
        if (this.month() === 11 && this.date() > 25) {
          var w = h(this).startOf(t).add(1, t).date(m), g = h(this).endOf(a);
          if (w.isBefore(g))
            return 1;
        }
        var O = h(this).startOf(t).date(m).startOf(a).subtract(1, "millisecond"), V = this.diff(O, a, !0);
        return V < 0 ? h(this).startOf("week").week() : Math.ceil(V);
      }, l.weeks = function(v) {
        return v === void 0 && (v = null), this.week(v);
      };
    };
  });
})(Lt);
var Uo = Lt.exports;
const zo = /* @__PURE__ */ Ce(Uo);
function je(e, r) {
  const a = De(e, r);
  if (!a)
    throw new Error(`Could not resolve ${e.description}`);
  return a;
}
const ft = Object.fromEntries(
  Object.entries(/* @__PURE__ */ Object.assign({ "../node_modules/dayjs/esm/locale/af.js": () => import("./af-3f5e3754.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/am.js": () => import("./am-bc833d79.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ar-dz.js": () => import("./ar-dz-2b677c27.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ar-iq.js": () => import("./ar-iq-9280b179.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ar-kw.js": () => import("./ar-kw-06673fb3.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ar-ly.js": () => import("./ar-ly-b364c556.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ar-ma.js": () => import("./ar-ma-e9b96f88.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ar-sa.js": () => import("./ar-sa-920b6966.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ar-tn.js": () => import("./ar-tn-5d2ebe87.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ar.js": () => import("./ar-a3aa818f.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/az.js": () => import("./az-659b56f9.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/be.js": () => import("./be-3b4f9783.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/bg.js": () => import("./bg-0672cd7d.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/bi.js": () => import("./bi-951682c2.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/bm.js": () => import("./bm-9d7e855b.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/bn-bd.js": () => import("./bn-bd-087a7a1c.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/bn.js": () => import("./bn-0c0acd44.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/bo.js": () => import("./bo-19632568.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/br.js": () => import("./br-5a3443b7.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/bs.js": () => import("./bs-ad641200.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ca.js": () => import("./ca-035ea682.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/cs.js": () => import("./cs-debeec9e.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/cv.js": () => import("./cv-dcf48c54.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/cy.js": () => import("./cy-daa2e33d.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/da.js": () => import("./da-dc449cac.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/de-at.js": () => import("./de-at-5acf665a.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/de-ch.js": () => import("./de-ch-6b981a67.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/de.js": () => import("./de-77586bc3.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/dv.js": () => import("./dv-65849a7f.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/el.js": () => import("./el-ae4ad393.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/en-au.js": () => import("./en-au-8c36e702.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/en-ca.js": () => import("./en-ca-c5437740.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/en-gb.js": () => import("./en-gb-c2cc134a.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/en-ie.js": () => import("./en-ie-d3ac9ac2.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/en-il.js": () => import("./en-il-6dd24280.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/en-in.js": () => import("./en-in-2f2879f3.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/en-nz.js": () => import("./en-nz-c996ce95.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/en-sg.js": () => import("./en-sg-278f7244.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/en-tt.js": () => import("./en-tt-dcca6678.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/en.js": () => import("./en-4402d6fc.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/eo.js": () => import("./eo-2b962c7e.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/es-do.js": () => import("./es-do-e5ec18dc.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/es-mx.js": () => import("./es-mx-0b0fdda9.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/es-pr.js": () => import("./es-pr-ecf92870.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/es-us.js": () => import("./es-us-9a974819.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/es.js": () => import("./es-542d397d.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/et.js": () => import("./et-cc745c6f.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/eu.js": () => import("./eu-1819a0bf.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/fa.js": () => import("./fa-759da5ca.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/fi.js": () => import("./fi-48c34162.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/fo.js": () => import("./fo-1a56e22a.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/fr-ca.js": () => import("./fr-ca-a08d1ab6.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/fr-ch.js": () => import("./fr-ch-9e54ac3f.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/fr.js": () => import("./fr-34da226b.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/fy.js": () => import("./fy-32e86ec3.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ga.js": () => import("./ga-5443425a.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/gd.js": () => import("./gd-525324a8.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/gl.js": () => import("./gl-ecd4c576.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/gom-latn.js": () => import("./gom-latn-0de894a4.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/gu.js": () => import("./gu-f8a9ff06.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/he.js": () => import("./he-c3d5738f.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/hi.js": () => import("./hi-06d9d378.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/hr.js": () => import("./hr-df6e22c2.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ht.js": () => import("./ht-560ce1fa.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/hu.js": () => import("./hu-36659a19.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/hy-am.js": () => import("./hy-am-ec1e6b6f.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/id.js": () => import("./id-e83ede43.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/is.js": () => import("./is-112d618e.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/it-ch.js": () => import("./it-ch-74dc20fb.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/it.js": () => import("./it-a9bef34d.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ja.js": () => import("./ja-81ac0bce.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/jv.js": () => import("./jv-eb80b191.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ka.js": () => import("./ka-408178cc.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/kk.js": () => import("./kk-7182d80c.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/km.js": () => import("./km-c8d90f37.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/kn.js": () => import("./kn-1ef13da8.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ko.js": () => import("./ko-d74dbac1.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ku.js": () => import("./ku-217c312b.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ky.js": () => import("./ky-9beeab3e.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/lb.js": () => import("./lb-bbb0769c.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/lo.js": () => import("./lo-c0a222fc.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/lt.js": () => import("./lt-7733040c.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/lv.js": () => import("./lv-8456bf8c.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/me.js": () => import("./me-60049fb4.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/mi.js": () => import("./mi-a00211ea.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/mk.js": () => import("./mk-750f2eb3.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ml.js": () => import("./ml-3d864495.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/mn.js": () => import("./mn-c3b569a5.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/mr.js": () => import("./mr-caa70638.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ms-my.js": () => import("./ms-my-9edfd210.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ms.js": () => import("./ms-8a0b04c8.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/mt.js": () => import("./mt-5924bb24.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/my.js": () => import("./my-7ec0e79b.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/nb.js": () => import("./nb-55474232.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ne.js": () => import("./ne-28b71d4d.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/nl-be.js": () => import("./nl-be-ab2f9375.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/nl.js": () => import("./nl-f2df7562.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/nn.js": () => import("./nn-ae0c69b8.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/oc-lnc.js": () => import("./oc-lnc-e86add7d.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/pa-in.js": () => import("./pa-in-58db4e88.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/pl.js": () => import("./pl-6123f464.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/pt-br.js": () => import("./pt-br-72da3648.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/pt.js": () => import("./pt-6d21f766.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/rn.js": () => import("./rn-fe91690b.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ro.js": () => import("./ro-f0333df1.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ru.js": () => import("./ru-8092165f.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/rw.js": () => import("./rw-8e49f17e.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/sd.js": () => import("./sd-f5f464cc.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/se.js": () => import("./se-d0247819.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/si.js": () => import("./si-23229411.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/sk.js": () => import("./sk-2ae651e5.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/sl.js": () => import("./sl-d651cb86.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/sq.js": () => import("./sq-851e451a.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/sr-cyrl.js": () => import("./sr-cyrl-150c337e.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/sr.js": () => import("./sr-26ffbdc9.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ss.js": () => import("./ss-70c27ddd.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/sv-fi.js": () => import("./sv-fi-53a8b8bd.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/sv.js": () => import("./sv-bf43bdc9.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/sw.js": () => import("./sw-3c86b419.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ta.js": () => import("./ta-dde447c0.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/te.js": () => import("./te-d039e67a.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/tet.js": () => import("./tet-2478e8c8.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/tg.js": () => import("./tg-374d7196.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/th.js": () => import("./th-cfb73f82.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/tk.js": () => import("./tk-6502e590.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/tl-ph.js": () => import("./tl-ph-f36e80af.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/tlh.js": () => import("./tlh-6d81a812.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/tr.js": () => import("./tr-1608d107.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/tzl.js": () => import("./tzl-e019f0a0.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/tzm-latn.js": () => import("./tzm-latn-4a3fedcb.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/tzm.js": () => import("./tzm-9a26d476.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ug-cn.js": () => import("./ug-cn-7370b4b7.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/uk.js": () => import("./uk-f2be452c.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/ur.js": () => import("./ur-5e01f781.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/uz-latn.js": () => import("./uz-latn-ec9b852e.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/uz.js": () => import("./uz-f44d7936.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/vi.js": () => import("./vi-8106a30d.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/x-pseudo.js": () => import("./x-pseudo-20ac0200.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/yo.js": () => import("./yo-45cb4db7.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/zh-cn.js": () => import("./zh-cn-7af2941c.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/zh-hk.js": () => import("./zh-hk-857b348c.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/zh-tw.js": () => import("./zh-tw-58dac95d.js").then((e) => e.default), "../node_modules/dayjs/esm/locale/zh.js": () => import("./zh-26803c4f.js").then((e) => e.default) })).map(
    ([e, r]) => {
      var a;
      return [(a = e.match(/([\w-]*)\.js$/)) == null ? void 0 : a[1], r];
    }
  )
), Wo = { class: "flex justify-between items-center px-2 py-1.5" }, Ko = { class: "flex-shrink-0" }, Go = { class: "inline-flex rounded-full" }, Zo = {
  class: "w-5 h-5",
  fill: "none",
  stroke: "currentColor",
  viewBox: "0 0 24 24",
  xmlns: "http://www.w3.org/2000/svg"
}, qo = ["d"], Jo = { class: "px-1.5 space-x-1.5 flex flex-1" }, Xo = { class: "flex-1 flex rounded-md" }, Qo = ["textContent"], er = { class: "flex-1 flex rounded-md" }, tr = ["textContent"], or = { class: "flex-shrink-0" }, rr = { class: "inline-flex rounded-full" }, ar = {
  class: "w-5 h-5",
  fill: "none",
  stroke: "currentColor",
  viewBox: "0 0 24 24",
  xmlns: "http://www.w3.org/2000/svg"
}, nr = ["d"], Ge = /* @__PURE__ */ ue({
  __name: "Header",
  props: {
    panel: {},
    calendar: {}
  },
  setup(e) {
    return (r, a) => (z(), Z("div", Wo, [
      A("div", Ko, [
        me(A("span", Go, [
          A("button", {
            type: "button",
            class: "p-1.5 rounded-full bg-white text-vtd-secondary-600 transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none dark:bg-vtd-secondary-800 dark:text-vtd-secondary-300 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-secondary-300 dark:focus:bg-vtd-secondary-600 dark:focus:text-vtd-secondary-100 dark:focus:border-vtd-primary-500 dark:focus:ring-opacity-25 dark:focus:bg-opacity-50",
            onClick: a[0] || (a[0] = (t) => r.panel.calendar ? r.calendar.onPrevious() : r.calendar.onPreviousYear())
          }, [
            (z(), Z("svg", Zo, [
              A("path", {
                "stroke-linecap": "round",
                "stroke-linejoin": "round",
                "stroke-width": "1.5",
                d: r.panel.calendar ? "M15 19l-7-7 7-7" : "M11 19l-7-7 7-7m8 14l-7-7 7-7"
              }, null, 8, qo)
            ]))
          ])
        ], 512), [
          [pe, r.panel.calendar || r.panel.year]
        ])
      ]),
      A("div", Jo, [
        A("span", Xo, [
          A("button", {
            type: "button",
            class: "px-3 py-1.5 block w-full leading-relaxed rounded-md bg-white text-xs 2xl:text-sm tracking-wide text-vtd-secondary-600 font-semibold sm:font-medium transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none uppercase dark:bg-vtd-secondary-800 dark:text-vtd-secondary-300 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-secondary-300 dark:focus:bg-vtd-secondary-600 dark:focus:text-vtd-secondary-100 dark:focus:border-vtd-primary-500 dark:focus:ring-opacity-25 dark:focus:bg-opacity-50",
            onClick: a[1] || (a[1] = (t) => r.calendar.openMonth()),
            textContent: ae(r.calendar.month)
          }, null, 8, Qo)
        ]),
        A("span", er, [
          A("button", {
            type: "button",
            class: "px-3 py-1.5 block w-full leading-relaxed rounded-md bg-white text-xs 2xl:text-sm tracking-wide text-vtd-secondary-600 font-semibold sm:font-medium transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none uppercase dark:bg-vtd-secondary-800 dark:text-vtd-secondary-300 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-secondary-300 dark:focus:bg-vtd-secondary-600 dark:focus:text-vtd-secondary-100 dark:focus:border-vtd-primary-500 dark:focus:ring-opacity-25 dark:focus:bg-opacity-50",
            onClick: a[2] || (a[2] = (t) => r.calendar.openYear()),
            textContent: ae(r.calendar.year)
          }, null, 8, tr)
        ])
      ]),
      A("div", or, [
        me(A("span", rr, [
          A("button", {
            type: "button",
            class: "p-1.5 rounded-full bg-white text-vtd-secondary-600 transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none dark:bg-vtd-secondary-800 dark:text-vtd-secondary-300 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-secondary-300 dark:focus:bg-vtd-secondary-600 dark:focus:text-vtd-secondary-100 dark:focus:border-vtd-primary-500 dark:focus:ring-opacity-25 dark:focus:bg-opacity-50",
            onClick: a[3] || (a[3] = (t) => r.panel.calendar ? r.calendar.onNext() : r.calendar.onNextYear())
          }, [
            (z(), Z("svg", ar, [
              A("path", {
                "stroke-linecap": "round",
                "stroke-linejoin": "round",
                "stroke-width": "1.5",
                d: r.panel.calendar ? "M9 5l7 7-7 7" : "M13 5l7 7-7 7M5 5l7 7-7 7"
              }, null, 8, nr)
            ]))
          ])
        ], 512), [
          [pe, r.panel.calendar || r.panel.year]
        ])
      ])
    ]));
  }
}), Ft = Symbol("isBetweenRange"), Bt = Symbol(
  "betweenRangeClasses"
), Nt = Symbol("datepickerClasses"), Rt = Symbol("atMouseOver"), It = Symbol("setToToday"), Ht = Symbol("setToYesterday"), Ut = Symbol("setToLastDay"), zt = Symbol("setToThisMonth"), Wt = Symbol("setToLastMonth"), Kt = Symbol("setToCustomShortcut"), sr = {
  key: 0,
  class: "relative w-full border-t border-b-0 sm:border-t-0 sm:border-b lg:border-b-0 lg:border-r border-black/[.1] order-last sm:order-none dark:border-vtd-secondary-700/[1] sm:mt-1 lg:mr-1 sm:mb-1 lg:mb-0 sm:mx-1 lg:mx-0 sm:w-auto"
}, lr = {
  key: 0,
  class: "grid grid-cols-2 sm:grid-cols-3 gap-1 lg:block w-full pr-5 sm:pr-6 mt-1.5 sm:mt-0 sm:mb-1.5 lg:mb-0"
}, ur = ["onClick", "textContent"], dr = {
  key: 1,
  class: "grid grid-cols-2 sm:grid-cols-3 gap-1 lg:block w-full pr-5 sm:pr-6 mt-1.5 sm:mt-0 sm:mb-1.5 lg:mb-0"
}, mt = /* @__PURE__ */ ue({
  __name: "Shortcut",
  props: {
    shortcuts: { type: [Boolean, Function] },
    close: { type: Function },
    asRange: { type: Boolean },
    asSingle: { type: Boolean },
    i18n: {}
  },
  setup(e) {
    const r = e, a = je(It), t = je(Ht), s = je(Ut), n = je(zt), h = je(Wt), l = je(Kt), v = le(() => typeof r.shortcuts == "function" ? r.shortcuts() : !1);
    return (m, w) => r.asRange && r.asSingle || r.asRange && !r.asSingle ? (z(), Z("div", sr, [
      v.value ? (z(), Z("ol", lr, [
        (z(!0), Z(we, null, We(v.value, (g, O) => (z(), Z("li", { key: O }, [
          A("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: $e((V) => se(l)(g, m.close), ["prevent"]),
            textContent: ae(g.label)
          }, null, 8, ur)
        ]))), 128))
      ])) : (z(), Z("ol", dr, [
        A("li", null, [
          A("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: w[0] || (w[0] = $e((g) => se(a)(m.close), ["prevent"]))
          }, ae(r.i18n.today), 1)
        ]),
        A("li", null, [
          A("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: w[1] || (w[1] = $e((g) => se(t)(m.close), ["prevent"]))
          }, ae(r.i18n.yesterday), 1)
        ]),
        A("li", null, [
          A("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: w[2] || (w[2] = $e((g) => se(s)(7, m.close), ["prevent"]))
          }, ae(r.i18n.past(7)), 1)
        ]),
        A("li", null, [
          A("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: w[3] || (w[3] = $e((g) => se(s)(30, m.close), ["prevent"]))
          }, ae(r.i18n.past(30)), 1)
        ]),
        A("li", null, [
          A("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: w[4] || (w[4] = $e((g) => se(n)(m.close), ["prevent"]))
          }, ae(r.i18n.currentMonth), 1)
        ]),
        A("li", null, [
          A("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: w[5] || (w[5] = $e((g) => se(h)(m.close), ["prevent"]))
          }, ae(r.i18n.pastMonth), 1)
        ])
      ]))
    ])) : ye("", !0);
  }
}), ir = { class: "grid grid-cols-7 gap-y-0.5 my-1" }, cr = {
  key: 0,
  class: "col-span-7 border-b relative dark:border-vtd-secondary-600"
}, fr = { class: "absolute -left-2 top-1/2 -translate-y-2/4 bg-white dark:bg-vtd-secondary-800 text-[8px] pr-2 text-vtd-secondary-400" }, mr = ["data-tooltip"], pr = ["disabled", "data-date", "onClick", "onMouseenter", "onFocusin", "textContent"], Ze = /* @__PURE__ */ ue({
  __name: "Calendar",
  props: {
    calendar: {},
    weeks: {},
    weekNumber: { type: Boolean },
    asRange: { type: Boolean }
  },
  emits: ["updateDate"],
  setup(e, { emit: r }) {
    const a = je(Ft), t = je(Bt), s = je(Nt), n = je(Rt);
    return (h, l) => (z(), Z("div", ir, [
      ee(Xt, {
        "enter-from-class": "opacity-0",
        "enter-to-class": "opacity-100",
        "enter-active-class": "transition-opacity ease-out duration-300",
        "leave-active-class": "transition-opacity ease-in duration-200",
        "leave-from-class": "opacity-100",
        "leave-to-class": "opacity-0"
      }, {
        default: Fe(() => [
          (z(!0), Z(we, null, We(h.calendar.date(), (v, m) => (z(), Z(we, { key: m }, [
            m % 7 === 0 && h.weekNumber ? (z(), Z("div", cr, [
              A("span", fr, ae(v.week()), 1)
            ])) : ye("", !0),
            A("div", {
              class: Te(["relative", { "vtd-tooltip": h.asRange && v.duration() }]),
              "data-tooltip": `${v.duration()}`
            }, [
              ee(vt, {
                "enter-from-class": "opacity-0",
                "enter-to-class": "opacity-100",
                "enter-active-class": "transition-opacity ease-out duration-200",
                "leave-active-class": "transition-opacity ease-in duration-150",
                "leave-from-class": "opacity-100",
                "leave-to-class": "opacity-0"
              }, {
                default: Fe(() => [
                  se(a)(v) || v.hovered() ? (z(), Z("span", {
                    key: 0,
                    class: Te(["absolute bg-vtd-primary-100 bg-opacity-60 dark:bg-vtd-secondary-700 dark:bg-opacity-50", se(t)(v)])
                  }, null, 2)) : ye("", !0)
                ]),
                _: 2
              }, 1024),
              A("button", {
                type: "button",
                class: Te(["vtd-datepicker-date relative w-[2.7rem] h-[2.7rem] lg:w-10 lg:h-10 flex justify-center items-center text-xs 2xl:text-sm", [
                  se(s)(v),
                  h.asRange ? "transition-all" : "transition-colors"
                ]]),
                disabled: v.disabled || v.inRange(),
                "data-date": v.toDate(),
                onClick: (w) => r("updateDate", v),
                onMouseenter: (w) => se(n)(v),
                onFocusin: (w) => se(n)(v),
                textContent: ae(v.date())
              }, null, 42, pr)
            ], 10, mr)
          ], 64))), 128))
        ]),
        _: 1
      })
    ]));
  }
}), vr = { class: "flex flex-wrap" }, hr = { class: "flex rounded-md mt-1.5" }, yr = ["onClick", "textContent"], qe = /* @__PURE__ */ ue({
  __name: "Year",
  props: {
    years: {}
  },
  emits: ["updateYear"],
  setup(e, { emit: r }) {
    return (a, t) => (z(), Z("div", vr, [
      (z(!0), Z(we, null, We(a.years, (s, n) => (z(), Z("div", {
        key: n,
        class: "w-1/2 px-0.5"
      }, [
        A("span", hr, [
          A("button", {
            type: "button",
            class: "px-3 py-2 block w-full leading-6 rounded-md bg-white text-xs 2xl:text-sm tracking-wide text-vtd-secondary-600 font-medium transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none uppercase dark:bg-vtd-secondary-800 dark:hover:bg-vtd-secondary-700 dark:text-vtd-secondary-300 dark:hover:text-vtd-secondary-100 dark:focus:bg-vtd-secondary-700",
            onClick: (h) => r("updateYear", s),
            textContent: ae(s)
          }, null, 8, yr)
        ])
      ]))), 128))
    ]));
  }
}), br = { class: "grid grid-cols-7 py-2 mt-0.5" }, gr = ["textContent"], Je = /* @__PURE__ */ ue({
  __name: "Week",
  props: {
    weeks: {}
  },
  setup(e) {
    return (r, a) => (z(), Z("div", br, [
      (z(!0), Z(we, null, We(r.weeks, (t, s) => (z(), Z("div", {
        key: s,
        class: "text-vtd-secondary-500 text-xs 2xl:text-sm tracking-wide font-medium text-center cursor-default dark:text-vtd-secondary-400"
      }, [
        A("span", {
          textContent: ae(t)
        }, null, 8, gr)
      ]))), 128))
    ]));
  }
}), xr = { class: "flex flex-wrap mt-1.5" }, wr = { class: "flex rounded-md mt-1.5" }, kr = ["onClick", "textContent"], Xe = /* @__PURE__ */ ue({
  __name: "Month",
  props: {
    months: {}
  },
  emits: ["updateMonth"],
  setup(e, { emit: r }) {
    return (a, t) => (z(), Z("div", xr, [
      (z(!0), Z(we, null, We(a.months, (s, n) => (z(), Z("div", {
        key: n,
        class: "w-1/2 px-0.5"
      }, [
        A("span", wr, [
          A("button", {
            type: "button",
            class: "px-3 py-2 block w-full leading-6 rounded-md bg-white text-xs 2xl:text-sm tracking-wide text-vtd-secondary-600 font-medium transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none uppercase dark:bg-vtd-secondary-800 dark:hover:bg-vtd-secondary-700 dark:text-vtd-secondary-300 dark:hover:text-vtd-secondary-100 dark:focus:bg-vtd-secondary-700",
            onClick: (h) => r("updateMonth", n),
            textContent: ae(s)
          }, null, 8, kr)
        ])
      ]))), 128))
    ]));
  }
});
function jr() {
  const e = (l) => {
    const v = [], m = l.localeData().firstDayOfWeek();
    for (let w = 0; w <= l.date(0 - m).day(); w++)
      v.push(l.date(0).subtract(w, "day"));
    return v.sort((w, g) => w.date() - g.date());
  };
  return {
    usePreviousDate: e,
    useCurrentDate: (l) => Array.from(
      {
        length: l.daysInMonth()
      },
      (v, m) => l.date(m + 1)
    ),
    useNextDate: (l) => {
      const v = [];
      for (let m = 1; m <= 42 - (e(l).length + l.daysInMonth()); m++)
        v.push(l.date(m).month(l.month()).add(1, "month"));
      return v;
    },
    useDisableDate: (l, { disableDate: v }) => typeof v == "function" ? v(l.toDate()) : !1,
    useBetweenRange: (l, { previous: v, next: m }) => {
      const w = v.isAfter(m, "date") ? "(]" : "[)";
      return !!(l.isBetween(v, m, "date", w) && !l.off);
    },
    useToValueFromString: (l, { formatter: v }) => l.format(v.date),
    useToValueFromArray: ({ previous: l, next: v }, {
      formatter: m,
      separator: w
    }) => `${l.format(m.date)}${w}${v.format(
      m.date
    )}`
  };
}
function _r() {
  return {
    useVisibleViewport: (r) => {
      if (r) {
        const { right: a } = r.getBoundingClientRect(), t = window.innerWidth || document.documentElement.clientWidth;
        return a > t;
      } else
        return null;
    }
  };
}
const $r = ["disabled", "placeholder", "onKeyup"], Mr = { class: "absolute inset-y-0 right-0 inline-flex items-center rounded-md overflow-hidden" }, Sr = ["disabled"], Dr = {
  class: "w-5 h-5",
  fill: "none",
  stroke: "currentColor",
  viewBox: "0 0 24 24",
  xmlns: "http://www.w3.org/2000/svg"
}, Or = {
  key: 0,
  "stroke-linecap": "round",
  "stroke-linejoin": "round",
  "stroke-width": "1.5",
  d: "M6 18L18 6M6 6l12 12"
}, Pr = {
  key: 1,
  "stroke-linecap": "round",
  "stroke-linejoin": "round",
  "stroke-width": "1.5",
  d: "M8 7V3m8 4V3m-9 8h10M5 21h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v12a2 2 0 002 2z"
}, Vr = { class: "flex flex-wrap lg:flex-nowrap" }, Tr = { class: "relative flex flex-wrap sm:flex-nowrap p-1 w-full" }, Ar = {
  key: 0,
  class: "hidden h-full absolute inset-0 sm:flex justify-center items-center"
}, Yr = /* @__PURE__ */ A("div", { class: "h-full border-r border-black/[.1] dark:border-vtd-secondary-700/[1]" }, null, -1), Cr = [
  Yr
], Er = { class: "px-0.5 sm:px-2" }, Lr = {
  key: 1,
  class: "relative w-full md:w-1/2 lg:w-80 overflow-hidden mt-3 sm:mt-0 sm:ml-2"
}, Fr = { class: "px-0.5 sm:px-2" }, Br = { key: 0 }, Nr = { class: "mt-2 mx-2 py-1.5 border-t border-black/[.1] dark:border-vtd-secondary-700/[1]" }, Rr = { class: "mt-1.5 sm:flex sm:flex-row-reverse" }, Ir = ["disabled", "onClick", "textContent"], Hr = ["onClick", "textContent"], Ur = {
  key: 1,
  class: "sm:hidden"
}, zr = { class: "mt-2 mx-2 py-1.5 border-t border-black/[.1] dark:border-vtd-secondary-700/[1]" }, Wr = { class: "mt-1.5 sm:flex sm:flex-row-reverse" }, Kr = ["onClick", "textContent"], Gr = {
  key: 1,
  class: "flex"
}, Zr = { class: "bg-white rounded-lg shadow-sm border border-black/[.1] px-3 py-3 sm:px-4 sm:py-4 dark:bg-vtd-secondary-800 dark:border-vtd-secondary-700/[1]" }, qr = { class: "flex flex-wrap lg:flex-nowrap" }, Jr = { class: "relative flex flex-wrap sm:flex-nowrap p-1 w-full" }, Xr = {
  key: 0,
  class: "hidden h-full absolute inset-0 sm:flex justify-center items-center"
}, Qr = /* @__PURE__ */ A("div", { class: "h-full border-r border-black/[.1] dark:border-vtd-secondary-700/[1]" }, null, -1), ea = [
  Qr
], ta = { class: "px-0.5 sm:px-2" }, oa = {
  key: 1,
  class: "relative w-full md:w-1/2 lg:w-80 overflow-hidden mt-3 sm:mt-0 sm:ml-2"
}, ra = { class: "px-0.5 sm:px-2" }, aa = { key: 0 }, na = { class: "mt-2 mx-2 py-1.5 border-t border-black/[.1] dark:border-vtd-secondary-700/[1]" }, sa = { class: "mt-1.5 sm:flex sm:flex-row-reverse" }, la = ["disabled", "textContent"], ua = /* @__PURE__ */ ue({
  __name: "VueTailwindDatePicker",
  props: {
    noInput: { type: Boolean },
    overlay: { type: Boolean },
    asSingle: { type: Boolean },
    useRange: { type: Boolean },
    placeholder: { default: "" },
    i18n: { default: "en" },
    inputClasses: { default: "" },
    disabled: { type: Boolean, default: !1 },
    disableInRange: { type: Boolean, default: !1 },
    disableDate: { type: [Boolean, Function], default: !1 },
    autoApply: { type: Boolean, default: !0 },
    shortcuts: { type: [Boolean, Function], default: !0 },
    separator: { default: " ~ " },
    formatter: { default: () => ({
      date: "YYYY-MM-DD HH:mm:ss",
      month: "MMM"
    }) },
    startFrom: { default: () => /* @__PURE__ */ new Date() },
    weekdaysSize: { default: "short" },
    weekNumber: { type: Boolean, default: !1 },
    options: { default: () => ({
      shortcuts: {
        today: "Today",
        yesterday: "Yesterday",
        past: (e) => `Last ${e} Days`,
        currentMonth: "This Month",
        pastMonth: "Last Month"
      },
      footer: {
        apply: "Apply",
        cancel: "Cancel"
      }
    }) },
    modelValue: { default: () => [/* @__PURE__ */ new Date(), /* @__PURE__ */ new Date()] }
  },
  emits: ["update:modelValue", "selectMonth", "selectYear", "selectRightMonth", "selectRightYear", "clickPrev", "clickNext", "clickRightPrev", "clickRightNext"],
  setup(e, { expose: r, emit: a }) {
    const t = e, {
      useCurrentDate: s,
      useDisableDate: n,
      useBetweenRange: h,
      useNextDate: l,
      usePreviousDate: v,
      useToValueFromArray: m,
      useToValueFromString: w
    } = jr(), { useVisibleViewport: g } = _r();
    i.extend(Ao), i.extend(Co), i.extend(Lo), i.extend(Bo), i.extend(Ro), i.extend(Ho), i.extend(zo);
    const O = J(null), V = J(null), B = J(null), x = J(""), P = J(null), k = J(""), M = J([]), y = J([]), N = J(null), L = J(null), d = pt({
      previous: {
        calendar: !0,
        month: !1,
        year: !1
      },
      next: {
        calendar: !0,
        month: !1,
        year: !1
      }
    }), o = J({
      previous: i(),
      next: i().add(1, "month"),
      year: {
        previous: i().year(),
        next: i().year()
      },
      weeks: t.weekdaysSize === "min" ? i.weekdaysMin() : i.weekdaysShort(),
      months: t.formatter.month === "MMM" ? i.monthsShort() : i.months()
    }), b = le(() => o.value.weeks), I = le(() => o.value.months), T = le(() => {
      const { previous: p, next: c, year: f } = se(o);
      return {
        previous: {
          date: () => v(p).concat(s(p)).concat(l(p)).map((u) => (Object.assign(u, {
            today: u.isToday(),
            active: p.month() === u.month(),
            off: p.month() !== u.month(),
            sunday: u.day() === 0,
            disabled: n(u, t) && !D(u),
            inRange: () => {
              if (t.asSingle && !t.useRange)
                return p.month() !== u.month();
            },
            hovered: () => j() && M.value.length > 1 ? (u.isBetween(
              M.value[0],
              M.value[1],
              "date",
              "()"
            ) || u.isBetween(
              M.value[1],
              M.value[0],
              "date",
              "(]"
            )) && p.month() === u.month() : !1,
            duration: () => !1
          }), u)),
          month: p && p.format(t.formatter.month),
          year: p && p.year(),
          years: () => Array.from(
            {
              length: 12
            },
            (u, _) => f.previous + _
          ),
          onPrevious: () => {
            o.value.previous = p.subtract(1, "month"), a("clickPrev", o.value.previous);
          },
          onNext: () => {
            o.value.previous = p.add(1, "month"), p.diff(c, "month") === -1 && (o.value.next = c.add(1, "month")), a("clickNext", o.value.previous);
          },
          onPreviousYear: () => {
            o.value.year.previous = o.value.year.previous - 12;
          },
          onNextYear: () => {
            o.value.year.previous = o.value.year.previous + 12;
          },
          openMonth: () => {
            d.previous.month = !d.previous.month, d.previous.year = !1, d.previous.calendar = !d.previous.month;
          },
          setMonth: (u) => {
            o.value.previous = p.month(u), d.previous.month = !d.previous.month, d.previous.year = !1, d.previous.calendar = !d.previous.month, a("selectMonth", o.value.previous), Ie(() => {
              (o.value.next.isSame(o.value.previous, "month") || o.value.next.isBefore(o.value.previous)) && (o.value.next = o.value.previous.add(1, "month")), o.value.year.next = o.value.next.year();
            });
          },
          openYear: () => {
            d.previous.year = !d.previous.year, d.previous.month = !1, d.previous.calendar = !d.previous.year;
          },
          setYear: (u) => {
            o.value.previous = p.year(u), d.previous.year = !d.previous.year, d.previous.calendar = !d.previous.year, a("selectYear", o.value.previous), Ie(() => {
              (o.value.next.isSame(o.value.previous, "month") || o.value.next.isBefore(o.value.previous)) && (o.value.next = o.value.previous.add(1, "month")), o.value.year.previous = o.value.previous.year(), o.value.year.next = o.value.next.year();
            });
          }
        },
        next: {
          date: () => v(c).concat(s(c)).concat(l(c)).map((u) => (Object.assign(u, {
            today: u.isToday(),
            active: c.month() === u.month(),
            off: c.month() !== u.month(),
            sunday: u.day() === 0,
            disabled: n(u, t) && !D(u),
            inRange: () => {
              if (t.asSingle && !t.useRange)
                return c.month() !== u.month();
            },
            hovered: () => M.value.length > 1 ? (u.isBetween(
              M.value[0],
              M.value[1],
              "date",
              "()"
            ) || u.isBetween(
              M.value[1],
              M.value[0],
              "date",
              "(]"
            )) && c.month() === u.month() : !1,
            duration: () => !1
          }), u)),
          month: c && c.format(t.formatter.month),
          year: c && c.year(),
          years: () => Array.from(
            {
              length: 12
            },
            (u, _) => f.next + _
          ),
          onPrevious: () => {
            o.value.next = c.subtract(1, "month"), c.diff(p, "month") === 1 && (o.value.previous = p.subtract(1, "month")), a("clickRightPrev", o.value.next);
          },
          onNext: () => {
            o.value.next = c.add(1, "month"), a("clickRightNext", o.value.next);
          },
          onPreviousYear: () => {
            o.value.year.next = o.value.year.next - 12;
          },
          onNextYear: () => {
            o.value.year.next = o.value.year.next + 12;
          },
          openMonth: () => {
            d.next.month = !d.next.month, d.next.year = !1, d.next.calendar = !d.next.month;
          },
          setMonth: (u) => {
            o.value.next = c.month(u), d.next.month = !d.next.month, d.next.year = !1, d.next.calendar = !d.next.month, a("selectRightMonth", o.value.next), Ie(() => {
              (o.value.previous.isSame(o.value.next, "month") || o.value.previous.isAfter(o.value.next)) && (o.value.previous = o.value.next.subtract(
                1,
                "month"
              )), o.value.year.previous = o.value.previous.year();
            });
          },
          openYear: () => {
            d.next.year = !d.next.year, d.next.month = !1, d.next.calendar = !d.next.year;
          },
          setYear: (u) => {
            o.value.next = c.year(u), d.next.year = !d.next.year, d.next.month = !1, d.next.calendar = !d.next.year, a("selectRightYear", o.value.next), Ie(() => {
              (o.value.previous.isSame(o.value.next, "month") || o.value.previous.isAfter(o.value.next)) && (o.value.previous = o.value.next.subtract(
                1,
                "month"
              )), o.value.year.previous = o.value.previous.year(), o.value.year.next = o.value.next.year();
            });
          }
        }
      };
    }), C = J(!1);
    setTimeout(() => {
      C.value = !0;
    }, 250);
    function K() {
      return i().localeData().firstDayOfWeek();
    }
    function q(p) {
      const c = [...p], f = c.shift();
      return [...c, f];
    }
    function j() {
      return !t.useRange && !t.asSingle ? !0 : !t.useRange && t.asSingle ? !1 : t.useRange && !t.asSingle ? !0 : !!(t.useRange && t.asSingle);
    }
    function D(p) {
      if (t.disableInRange || k.value === "")
        return !1;
      let c, f;
      if (Array.isArray(t.modelValue)) {
        const [u, _] = t.modelValue;
        c = u, f = _;
      } else if (typeof t.modelValue == "object") {
        if (t.modelValue) {
          const [u, _] = Object.values(t.modelValue);
          c = u, f = _;
        }
      } else {
        const [u, _] = t.modelValue.split(t.separator);
        c = u, f = _;
      }
      return p.isBetween(
        i(c, t.formatter.date, !0),
        i(f, t.formatter.date, !0),
        "date",
        "[]"
      );
    }
    function $() {
      N.value = null, L.value = null, M.value = [], P.value = null;
    }
    function Y() {
      if (k.value = "", Array.isArray(t.modelValue))
        a("update:modelValue", []);
      else if (typeof t.modelValue == "object") {
        const p = {}, [c, f] = Object.keys(t.modelValue);
        p[c] = "", p[f] = "", a("update:modelValue", p);
      } else
        a("update:modelValue", "");
      y.value = [], V.value && V.value.focus();
    }
    r({ clearPicker: Y });
    function S() {
      if (j()) {
        const [p, c] = k.value.split(t.separator), [f, u] = [
          i(p, t.formatter.date, !0),
          i(c, t.formatter.date, !0)
        ];
        if (f.isValid() && u.isValid())
          if (E(f), E(u), Array.isArray(t.modelValue))
            a("update:modelValue", [p, c]);
          else if (typeof t.modelValue == "object") {
            const _ = {}, [F, W] = Object.keys(t.modelValue);
            _[F] = p, _[W] = c, a("update:modelValue", _);
          } else
            a(
              "update:modelValue",
              m(
                {
                  previous: f,
                  next: u
                },
                t
              )
            );
      } else {
        const p = i(k.value, t.formatter.date, !0);
        if (p.isValid())
          if (E(p), Array.isArray(t.modelValue))
            a("update:modelValue", [k.value]);
          else if (typeof t.modelValue == "object") {
            const c = {}, [f] = Object.keys(t.modelValue);
            c[f] = k.value, a("update:modelValue", c);
          } else
            a("update:modelValue", k.value);
      }
    }
    function E(p, c) {
      if (j())
        if (N.value)
          if (L.value = p, t.autoApply) {
            p.isBefore(N.value) ? k.value = m(
              {
                previous: p,
                next: N.value
              },
              t
            ) : k.value = m(
              {
                previous: N.value,
                next: p
              },
              t
            );
            const [f, u] = k.value.split(t.separator);
            if (Array.isArray(t.modelValue))
              a("update:modelValue", [
                i(f, t.formatter.date, !0).format(t.formatter.date),
                i(u, t.formatter.date, !0).format(t.formatter.date)
              ]);
            else if (typeof t.modelValue == "object") {
              const _ = {}, [F, W] = Object.keys(t.modelValue);
              _[F] = f, _[W] = u, a("update:modelValue", _);
            } else
              a(
                "update:modelValue",
                m(
                  {
                    previous: i(f, t.formatter.date, !0),
                    next: i(u, t.formatter.date, !0)
                  },
                  t
                )
              );
            c && c(), y.value = [], i(f, t.formatter.date, !0).isSame(
              i(u, t.formatter.date, !0),
              "month"
            ) || (o.value.previous = i(f, t.formatter.date, !0), o.value.next = i(u, t.formatter.date, !0)), $();
          } else {
            N.value.isAfter(p, "month") ? y.value = [p, N.value] : y.value = [N.value, p];
            const [f, u] = y.value;
            f.isSame(u, "month") || (o.value.previous = f, o.value.next = u), $();
          }
        else
          y.value = [], N.value = p, P.value = p, M.value.push(p), y.value.push(p), o.value.previous = p, o.value.next.isSame(p, "month") && (o.value.previous = o.value.next, o.value.next = p.add(1, "month"));
      else if (t.autoApply) {
        if (k.value = w(p, t), Array.isArray(t.modelValue))
          a("update:modelValue", [k.value]);
        else if (typeof t.modelValue == "object") {
          const f = {}, [u] = Object.keys(t.modelValue);
          f[u] = k.value, a("update:modelValue", f);
        } else
          a("update:modelValue", k.value);
        c && c(), y.value = [], $();
      } else
        y.value = [p], $();
    }
    function R(p) {
      if (y.value.length < 1)
        return !1;
      let c;
      if (j()) {
        const [f, u] = y.value;
        u.isBefore(f) ? c = m(
          {
            previous: u,
            next: f
          },
          t
        ) : c = m(
          {
            previous: f,
            next: u
          },
          t
        );
      } else {
        const [f] = y.value;
        c = f;
      }
      if (j()) {
        const [f, u] = c.split(t.separator);
        if (Array.isArray(t.modelValue))
          a("update:modelValue", [
            i(f, t.formatter.date, !0).format(t.formatter.date),
            i(u, t.formatter.date, !0).format(t.formatter.date)
          ]);
        else if (typeof t.modelValue == "object") {
          const _ = {}, [F, W] = Object.keys(t.modelValue);
          _[F] = f, _[W] = u, a("update:modelValue", _);
        } else
          a(
            "update:modelValue",
            m(
              {
                previous: i(f, t.formatter.date, !0),
                next: i(u, t.formatter.date, !0)
              },
              t
            )
          );
        k.value = c;
      } else if (k.value = c.format(t.formatter.date), Array.isArray(t.modelValue))
        a("update:modelValue", [k.value]);
      else if (typeof t.modelValue == "object") {
        const f = {}, [u] = Object.keys(t.modelValue);
        f[u] = k.value, a("update:modelValue", f);
      } else
        a("update:modelValue", k.value);
      p && p();
    }
    function G(p) {
      if (!j())
        return !1;
      if (N.value)
        M.value = [N.value, p];
      else
        return M.value = [], !1;
    }
    function Q(p) {
      if (N.value && t.autoApply)
        return !1;
      let c, f;
      if (M.value.length > 1) {
        const [u, _] = M.value;
        c = i(u, t.formatter.date, !0), f = i(_, t.formatter.date, !0);
      } else if (Array.isArray(t.modelValue))
        if (t.autoApply) {
          const [u, _] = t.modelValue;
          c = u && i(u, t.formatter.date, !0), f = _ && i(_, t.formatter.date, !0);
        } else {
          const [u, _] = y.value;
          c = i(u, t.formatter.date, !0), f = i(_, t.formatter.date, !0);
        }
      else if (typeof t.modelValue == "object")
        if (t.autoApply) {
          if (t.modelValue) {
            const [u, _] = Object.values(t.modelValue);
            c = u && i(u, t.formatter.date, !0), f = _ && i(_, t.formatter.date, !0);
          }
        } else {
          const [u, _] = y.value;
          c = i(u, t.formatter.date, !0), f = i(_, t.formatter.date, !0);
        }
      else if (t.autoApply) {
        const [u, _] = t.modelValue ? t.modelValue.split(t.separator) : [null, null];
        c = u && i(u, t.formatter.date, !0), f = _ && i(_, t.formatter.date, !0);
      } else {
        const [u, _] = y.value;
        c = i(u, t.formatter.date, !0), f = i(_, t.formatter.date, !0);
      }
      return c && f ? h(p, {
        previous: c,
        next: f
      }) : !1;
    }
    function te(p) {
      const { today: c, active: f, off: u, disabled: _ } = p;
      let F, W, ie;
      if (j())
        if (Array.isArray(t.modelValue))
          if (P.value) {
            const [U, X] = M.value;
            W = U && i(U, t.formatter.date, !0), ie = X && i(X, t.formatter.date, !0);
          } else if (t.autoApply) {
            const [U, X] = t.modelValue;
            W = U && i(U, t.formatter.date, !0), ie = X && i(X, t.formatter.date, !0);
          } else {
            const [U, X] = y.value;
            W = U && i(U, t.formatter.date, !0), ie = X && i(X, t.formatter.date, !0);
          }
        else if (typeof t.modelValue == "object")
          if (P.value) {
            const [U, X] = M.value;
            W = U && i(U, t.formatter.date, !0), ie = X && i(X, t.formatter.date, !0);
          } else if (t.autoApply) {
            const [U, X] = t.modelValue ? Object.values(t.modelValue) : [null, null];
            W = U && i(U, t.formatter.date, !0), ie = X && i(X, t.formatter.date, !0);
          } else {
            const [U, X] = y.value;
            W = U && i(U, t.formatter.date, !0), ie = X && i(X, t.formatter.date, !0);
          }
        else if (P.value) {
          const [U, X] = M.value;
          W = U && i(U, t.formatter.date, !0), ie = X && i(X, t.formatter.date, !0);
        } else if (t.autoApply) {
          const [U, X] = t.modelValue ? t.modelValue.split(t.separator) : [null, null];
          W = U && i(U, t.formatter.date, !0), ie = X && i(X, t.formatter.date, !0);
        } else {
          const [U, X] = y.value;
          W = U && i(U, t.formatter.date, !0), ie = X && i(X, t.formatter.date, !0);
        }
      else if (Array.isArray(t.modelValue))
        if (t.autoApply) {
          if (t.modelValue.length > 0) {
            const [U] = t.modelValue;
            W = i(U, t.formatter.date, !0);
          }
        } else {
          const [U] = y.value;
          W = U && i(U, t.formatter.date, !0);
        }
      else if (typeof t.modelValue == "object")
        if (t.autoApply) {
          if (t.modelValue) {
            const [U] = Object.values(t.modelValue);
            W = i(U, t.formatter.date, !0);
          }
        } else {
          const [U] = y.value;
          W = U && i(U, t.formatter.date, !0);
        }
      else if (t.autoApply) {
        if (t.modelValue) {
          const [U] = t.modelValue.split(t.separator);
          W = i(U, t.formatter.date, !0);
        }
      } else {
        const [U] = y.value;
        W = U && i(U, t.formatter.date, !0);
      }
      return f && (F = c ? "text-vtd-primary-500 font-semibold dark:text-vtd-primary-400 rounded-full focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none dark:bg-vtd-secondary-800 dark:text-vtd-secondary-300 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-secondary-300 dark:focus:bg-vtd-secondary-600 dark:focus:text-vtd-secondary-100 dark:focus:border-vtd-primary-500 dark:focus:ring-opacity-25 dark:focus:bg-opacity-50" : _ ? "text-vtd-secondary-600 font-normal disabled:text-vtd-secondary-500 disabled:cursor-not-allowed rounded-full" : p.isBetween(W, ie, "date", "()") ? "text-vtd-secondary-700 font-medium dark:text-vtd-secondary-100 rounded-full" : "text-vtd-secondary-600 font-medium dark:text-vtd-secondary-200 rounded-full"), u && (F = "text-vtd-secondary-400 font-light disabled:cursor-not-allowed"), W && ie && !u ? (p.isSame(W, "date") && (F = ie.isAfter(W, "date") ? "bg-vtd-primary-500 text-white font-bold rounded-l-full disabled:cursor-not-allowed" : "bg-vtd-primary-500 text-white font-bold rounded-r-full disabled:cursor-not-allowed", W.isSame(ie, "date") && (F = "bg-vtd-primary-500 text-white font-bold rounded-full disabled:cursor-not-allowed")), p.isSame(ie, "date") && (F = ie.isAfter(W, "date") ? "bg-vtd-primary-500 text-white font-bold rounded-r-full disabled:cursor-not-allowed" : "bg-vtd-primary-500 text-white font-bold rounded-l-full disabled:cursor-not-allowed", W.isSame(ie, "date") && (F = "bg-vtd-primary-500 text-white font-bold rounded-full disabled:cursor-not-allowed"))) : W && p.isSame(W, "date") && !u && (F = "bg-vtd-primary-500 text-white font-bold rounded-full disabled:cursor-not-allowed"), F;
    }
    function oe(p) {
      let c, f, u;
      if (c = "", !j())
        return c;
      if (Array.isArray(t.modelValue))
        if (M.value.length > 1) {
          const [_, F] = M.value;
          f = _ && i(_, t.formatter.date, !0), u = F && i(F, t.formatter.date, !0);
        } else if (t.autoApply) {
          const [_, F] = t.modelValue;
          f = _ && i(_, t.formatter.date, !0), u = F && i(F, t.formatter.date, !0);
        } else {
          const [_, F] = y.value;
          f = _ && i(_, t.formatter.date, !0), u = F && i(F, t.formatter.date, !0);
        }
      else if (typeof t.modelValue == "object")
        if (M.value.length > 1) {
          const [_, F] = M.value;
          f = _ && i(_, t.formatter.date, !0), u = F && i(F, t.formatter.date, !0);
        } else if (t.autoApply) {
          if (t.modelValue) {
            const [_, F] = Object.values(t.modelValue);
            f = _ && i(_, t.formatter.date, !0), u = F && i(F, t.formatter.date, !0);
          }
        } else {
          const [_, F] = y.value;
          f = _ && i(_, t.formatter.date, !0), u = F && i(F, t.formatter.date, !0);
        }
      else if (M.value.length > 1) {
        const [_, F] = M.value;
        f = _ && i(_, t.formatter.date, !0), u = F && i(F, t.formatter.date, !0);
      } else if (t.autoApply) {
        const [_, F] = t.modelValue ? t.modelValue.split(t.separator) : [null, null];
        f = _ && i(_, t.formatter.date, !0), u = F && i(F, t.formatter.date, !0);
      } else {
        const [_, F] = y.value;
        f = _ && i(_, t.formatter.date, !0), u = F && i(F, t.formatter.date, !0);
      }
      return f && u && (p.isSame(f, "date") ? (u.isBefore(f) && (c += " rounded-r-full inset-0"), f.isBefore(u) && (c += " rounded-l-full inset-0")) : p.isSame(u, "date") ? (u.isBefore(f) && (c += " rounded-l-full inset-0"), f.isBefore(u) && (c += " rounded-r-full inset-0")) : c += " inset-0"), c;
    }
    function de(p, c) {
      o.value.previous = i(p, t.formatter.date, !0), o.value.next = i(c, t.formatter.date, !0), (i.duration(o.value.next.diff(o.value.previous)).get("months") === 2 || i.duration(o.value.next.diff(o.value.previous)).get("months") === 1 && i.duration(o.value.next.diff(o.value.previous)).get("days") === 7) && (o.value.next = o.value.next.subtract(1, "month")), (o.value.next.isSame(o.value.previous, "month") || o.value.next.isBefore(o.value.previous)) && (o.value.next = o.value.previous.add(1, "month"));
    }
    function ce(p, c) {
      if (j())
        if (t.autoApply) {
          if (Array.isArray(t.modelValue))
            a("update:modelValue", [p, c]);
          else if (typeof t.modelValue == "object") {
            const f = {}, [u, _] = Object.keys(t.modelValue);
            f[u] = p, f[_] = c, a("update:modelValue", f);
          } else
            a(
              "update:modelValue",
              m(
                {
                  previous: i(p, t.formatter.date, !0),
                  next: i(c, t.formatter.date, !0)
                },
                t
              )
            );
          k.value = `${p}${t.separator}${c}`;
        } else
          y.value = [
            i(p, t.formatter.date, !0),
            i(c, t.formatter.date, !0)
          ];
      else if (t.autoApply) {
        if (Array.isArray(t.modelValue))
          a("update:modelValue", [p]);
        else if (typeof t.modelValue == "object") {
          const f = {}, [u] = Object.keys(t.modelValue);
          f[u] = p, a("update:modelValue", f);
        } else
          a("update:modelValue", p);
        k.value = p;
      } else
        y.value = [
          i(p, t.formatter.date, !0),
          i(c, t.formatter.date, !0)
        ];
      de(p, c);
    }
    function be(p) {
      const c = i().format(t.formatter.date), f = i().format(t.formatter.date);
      ce(c, f), p && p();
    }
    function ge(p) {
      const c = i().subtract(1, "day").format(t.formatter.date), f = i().subtract(1, "day").format(t.formatter.date);
      ce(c, f), p && p();
    }
    function he(p, c) {
      const f = i().subtract(p - 1, "day").format(t.formatter.date), u = i().format(t.formatter.date);
      ce(f, u), c && c();
    }
    function re(p) {
      const c = i().date(1).format(t.formatter.date), f = i().date(i().daysInMonth()).format(t.formatter.date);
      ce(c, f), p && p();
    }
    function fe(p) {
      const c = i().date(1).subtract(1, "month").format(t.formatter.date), f = i().date(0).format(t.formatter.date);
      ce(c, f), p && p();
    }
    function xe(p, c) {
      const [f, u] = p.atClick(), _ = i(f).format(t.formatter.date), F = i(u).format(t.formatter.date);
      ce(_, F), c && c();
    }
    Qt(
      () => y.value,
      (p) => {
        p.length > 0 && (d.previous.calendar = !0, d.previous.month = !1, d.previous.year = !1, d.next.calendar = !0, d.next.month = !1, d.next.year = !1);
      }
    ), _e(() => {
      t.placeholder ? x.value = t.placeholder : j() ? x.value = `${t.formatter.date}${t.separator}${t.formatter.date}` : x.value = t.formatter.date;
    }), _e(() => {
      const p = t.i18n, c = t.modelValue;
      Ie(async () => {
        if (p in ft) {
          const F = await ft[p]();
          i.locale(F, void 0, !0), i.locale(p);
        }
        let f, u;
        if (j()) {
          if (Array.isArray(c)) {
            if (c.length > 0) {
              const [F, W] = c;
              f = i(F, t.formatter.date, !0), u = i(W, t.formatter.date, !0);
            }
          } else if (typeof c == "object") {
            if (!eo(c))
              try {
                Object.keys(c);
              } catch {
                console.warn(
                  "[Vue Tailwind Datepicker]: It looks like you want to use Object as the argument %cv-model",
                  "font-style: italic; color: #42b883;",
                  ", but you pass it undefined or null."
                ), console.warn(
                  "[Vue Tailwind Datepicker]: We has replace with %c{ startDate: '', endDate: '' }",
                  "font-style: italic; color: #42b883;",
                  ", but you can replace manually."
                ), a("update:modelValue", {
                  startDate: "",
                  endDate: ""
                });
              }
            if (c) {
              const [F, W] = Object.values(c);
              f = F && i(F, t.formatter.date, !0), u = W && i(W, t.formatter.date, !0);
            }
          } else if (c) {
            const [F, W] = c.split(t.separator);
            f = i(F, t.formatter.date, !0), u = i(W, t.formatter.date, !0);
          }
          f && u ? (k.value = m(
            {
              previous: f,
              next: u
            },
            t
          ), u.isBefore(f, "month") ? (o.value.previous = u, o.value.next = f, o.value.year.previous = u.year(), o.value.year.next = f.year()) : u.isSame(f, "month") ? (o.value.previous = f, o.value.next = u.add(1, "month"), o.value.year.previous = f.year(), o.value.year.next = f.add(1, "year").year()) : (o.value.previous = f, o.value.next = u, o.value.year.previous = f.year(), o.value.year.next = u.year()), t.autoApply || (y.value = [f, u])) : (o.value.previous = i(t.startFrom), o.value.next = i(t.startFrom).add(1, "month"), o.value.year.previous = o.value.previous.year(), o.value.year.next = o.value.next.year());
        } else {
          if (Array.isArray(c)) {
            if (c.length > 0) {
              const [F] = c;
              f = i(F, t.formatter.date, !0);
            }
          } else if (typeof c == "object") {
            if (c) {
              const [F] = Object.values(c);
              f = i(F, t.formatter.date, !0);
            }
          } else if (c.length) {
            const [F] = c.split(t.separator);
            f = i(F, t.formatter.date, !0);
          }
          f && f.isValid() ? (k.value = w(f, t), o.value.previous = f, o.value.next = f.add(1, "month"), o.value.year.previous = f.year(), o.value.year.next = f.add(1, "year").year(), t.autoApply || (y.value = [f])) : (o.value.previous = i(t.startFrom), o.value.next = i(t.startFrom).add(1, "month"), o.value.year.previous = o.value.previous.year(), o.value.year.next = o.value.next.year());
        }
        const _ = t.weekdaysSize === "min" ? i.weekdaysMin() : i.weekdaysShort();
        o.value.weeks = K() ? q(_) : _, o.value.months = t.formatter.month === "MMM" ? i.monthsShort() : i.months();
      });
    });
    function ke(p) {
      return p && B.value === null && (B.value = g(O.value)), p && B.value ? "place-right" : "place-left";
    }
    function Re(p) {
      return p && B.value === null && (B.value = g(O.value)), B.value ? "left-auto right-0" : "left-0 right-auto";
    }
    return ne(Ft, Q), ne(Bt, oe), ne(Nt, te), ne(Rt, G), ne(It, be), ne(Ht, ge), ne(Ut, he), ne(zt, re), ne(Wt, fe), ne(Kt, xe), (p, c) => t.noInput ? C.value ? (z(), Z("div", Gr, [
      A("div", Zr, [
        A("div", qr, [
          t.shortcuts ? (z(), He(mt, {
            key: 0,
            shortcuts: t.shortcuts,
            "as-range": j(),
            "as-single": t.asSingle,
            i18n: t.options.shortcuts
          }, null, 8, ["shortcuts", "as-range", "as-single", "i18n"])) : ye("", !0),
          A("div", Jr, [
            j() && !t.asSingle ? (z(), Z("div", Xr, ea)) : ye("", !0),
            A("div", {
              class: Te(["relative w-full lg:w-80", {
                "mb-3 sm:mb-0 sm:mr-2 md:w-1/2": j() && !t.asSingle
              }])
            }, [
              ee(Ge, {
                panel: d.previous,
                calendar: T.value.previous
              }, null, 8, ["panel", "calendar"]),
              A("div", ta, [
                me(ee(Xe, {
                  months: I.value,
                  onUpdateMonth: T.value.previous.setMonth
                }, null, 8, ["months", "onUpdateMonth"]), [
                  [pe, d.previous.month]
                ]),
                me(ee(qe, {
                  years: T.value.previous.years(),
                  onUpdateYear: T.value.previous.setYear
                }, null, 8, ["years", "onUpdateYear"]), [
                  [pe, d.previous.year]
                ]),
                me(A("div", null, [
                  ee(Je, { weeks: b.value }, null, 8, ["weeks"]),
                  ee(Ze, {
                    calendar: T.value.previous,
                    weeks: b.value,
                    "as-range": j(),
                    "week-number": p.weekNumber,
                    onUpdateDate: c[3] || (c[3] = (f) => E(f))
                  }, null, 8, ["calendar", "weeks", "as-range", "week-number"])
                ], 512), [
                  [pe, d.previous.calendar]
                ])
              ])
            ], 2),
            j() && !t.asSingle ? (z(), Z("div", oa, [
              ee(Ge, {
                "as-prev-or-next": "",
                panel: d.next,
                calendar: T.value.next
              }, null, 8, ["panel", "calendar"]),
              A("div", ra, [
                me(ee(Xe, {
                  months: I.value,
                  onUpdateMonth: T.value.next.setMonth
                }, null, 8, ["months", "onUpdateMonth"]), [
                  [pe, d.next.month]
                ]),
                me(ee(qe, {
                  "as-prev-or-next": "",
                  years: T.value.next.years(),
                  onUpdateYear: T.value.next.setYear
                }, null, 8, ["years", "onUpdateYear"]), [
                  [pe, d.next.year]
                ]),
                me(A("div", null, [
                  ee(Je, { weeks: b.value }, null, 8, ["weeks"]),
                  ee(Ze, {
                    "as-prev-or-next": "",
                    calendar: T.value.next,
                    weeks: b.value,
                    "as-range": j(),
                    "week-number": p.weekNumber,
                    onUpdateDate: c[4] || (c[4] = (f) => E(f))
                  }, null, 8, ["calendar", "weeks", "as-range", "week-number"])
                ], 512), [
                  [pe, d.next.calendar]
                ])
              ])
            ])) : ye("", !0)
          ])
        ]),
        t.autoApply ? ye("", !0) : (z(), Z("div", aa, [
          A("div", na, [
            A("div", sa, [
              A("button", {
                type: "button",
                class: "away-apply-picker w-full transition ease-out duration-300 inline-flex justify-center rounded-md border border-transparent shadow-sm px-4 py-2 bg-vtd-primary-600 text-base font-medium text-white hover:bg-vtd-primary-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-vtd-primary-500 sm:ml-3 sm:w-auto sm:text-sm dark:ring-offset-vtd-secondary-800 disabled:cursor-not-allowed",
                disabled: t.asSingle ? y.value.length < 1 : y.value.length < 2,
                onClick: c[5] || (c[5] = (f) => R()),
                textContent: ae(t.options.footer.apply)
              }, null, 8, la)
            ])
          ])
        ]))
      ])
    ])) : ye("", !0) : (z(), He(se(Ot), {
      key: 0,
      id: "vtd",
      as: "div",
      class: "relative w-full"
    }, {
      default: Fe(({ open: f }) => [
        t.overlay && !t.disabled ? (z(), He(se(Oo), {
          key: 0,
          class: "fixed inset-0 bg-black opacity-30"
        })) : ye("", !0),
        ee(se(Do), {
          as: "label",
          class: "relative block"
        }, {
          default: Fe(() => [
            dt(p.$slots, "default", {
              value: k.value,
              placeholder: x.value,
              clear: Y
            }, () => [
              me(A("input", to({
                ref_key: "VtdInputRef",
                ref: V
              }, p.$attrs, {
                "onUpdate:modelValue": c[0] || (c[0] = (u) => k.value = u),
                type: "text",
                class: ["relative block w-full", [
                  t.disabled ? "cursor-default opacity-50" : "opacity-100",
                  p.inputClasses || "pl-3 pr-12 py-2.5 rounded-lg overflow-hidden border-solid text-sm text-vtd-secondary-700 placeholder-vtd-secondary-400 transition-colors bg-white border border-vtd-secondary-300 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none dark:bg-vtd-secondary-800 dark:border-vtd-secondary-700 dark:text-vtd-secondary-100 dark:placeholder-vtd-secondary-500 dark:focus:border-vtd-primary-500 dark:focus:ring-opacity-20"
                ]],
                disabled: t.disabled,
                autocomplete: "off",
                "data-lpignore": "true",
                "data-form-type": "other",
                placeholder: x.value,
                onKeyup: $e(S, ["stop"]),
                onKeydown: c[1] || (c[1] = $e(() => {
                }, ["stop"]))
              }), null, 16, $r), [
                [oo, k.value]
              ]),
              A("div", Mr, [
                A("button", {
                  type: "button",
                  disabled: t.disabled,
                  class: Te([
                    t.disabled ? "cursor-default opacity-50" : "opacity-100",
                    "px-2 py-1 mr-1 focus:outline-none text-vtd-secondary-400 dark:text-opacity-70 rounded-md"
                  ]),
                  onClick: c[2] || (c[2] = (u) => {
                    var _;
                    return t.disabled ? !1 : k.value ? Y() : (_ = V.value) == null ? void 0 : _.focus();
                  })
                }, [
                  dt(p.$slots, "inputIcon", { value: k.value }, () => [
                    (z(), Z("svg", Dr, [
                      k.value ? (z(), Z("path", Or)) : (z(), Z("path", Pr))
                    ]))
                  ])
                ], 10, Sr)
              ])
            ])
          ]),
          _: 3
        }),
        ee(vt, {
          "enter-from-class": "opacity-0 translate-y-3",
          "enter-to-class": "opacity-100 translate-y-0",
          "enter-active-class": "transform transition ease-out duration-200",
          "leave-active-class": "transform transition ease-in duration-150",
          "leave-from-class": "opacity-100 translate-y-0",
          "leave-to-class": "opacity-0 translate-y-3"
        }, {
          default: Fe(() => [
            t.disabled ? ye("", !0) : (z(), He(se(Po), {
              key: 0,
              as: "div",
              class: "relative z-50"
            }, {
              default: Fe(({ close: u }) => [
                A("div", {
                  class: Te(["absolute z-50 top-full sm:mt-2.5", Re(f)])
                }, [
                  A("div", {
                    ref_key: "VtdRef",
                    ref: O,
                    class: "fixed inset-0 z-50 overflow-y-auto sm:overflow-visible sm:static sm:z-auto bg-white dark:bg-vtd-secondary-800 sm:rounded-lg shadow-sm"
                  }, [
                    A("div", {
                      class: Te(["vtd-datepicker static sm:relative w-full bg-white sm:rounded-lg sm:shadow-sm border-0 sm:border border-black/[.1] px-3 py-3 sm:px-4 sm:py-4 dark:bg-vtd-secondary-800 dark:border-vtd-secondary-700/[1]", ke(f)])
                    }, [
                      A("div", Vr, [
                        t.shortcuts ? (z(), He(mt, {
                          key: 0,
                          shortcuts: t.shortcuts,
                          "as-range": j(),
                          "as-single": t.asSingle,
                          i18n: t.options.shortcuts,
                          close: u
                        }, null, 8, ["shortcuts", "as-range", "as-single", "i18n", "close"])) : ye("", !0),
                        A("div", Tr, [
                          j() && !t.asSingle ? (z(), Z("div", Ar, Cr)) : ye("", !0),
                          A("div", {
                            class: Te(["relative", {
                              "mb-3 sm:mb-0 sm:mr-2 w-full md:w-1/2 lg:w-80": j() && !t.asSingle,
                              "w-full": !j() && t.asSingle
                            }])
                          }, [
                            ee(Ge, {
                              panel: d.previous,
                              calendar: T.value.previous
                            }, null, 8, ["panel", "calendar"]),
                            A("div", Er, [
                              me(ee(Xe, {
                                months: I.value,
                                onUpdateMonth: T.value.previous.setMonth
                              }, null, 8, ["months", "onUpdateMonth"]), [
                                [pe, d.previous.month]
                              ]),
                              me(ee(qe, {
                                years: T.value.previous.years(),
                                onUpdateYear: T.value.previous.setYear
                              }, null, 8, ["years", "onUpdateYear"]), [
                                [pe, d.previous.year]
                              ]),
                              me(A("div", null, [
                                ee(Je, { weeks: b.value }, null, 8, ["weeks"]),
                                ee(Ze, {
                                  calendar: T.value.previous,
                                  weeks: b.value,
                                  "as-range": j(),
                                  "week-number": p.weekNumber,
                                  onUpdateDate: (_) => E(_, u)
                                }, null, 8, ["calendar", "weeks", "as-range", "week-number", "onUpdateDate"])
                              ], 512), [
                                [pe, d.previous.calendar]
                              ])
                            ])
                          ], 2),
                          j() && !t.asSingle ? (z(), Z("div", Lr, [
                            ee(Ge, {
                              "as-prev-or-next": "",
                              panel: d.next,
                              calendar: T.value.next
                            }, null, 8, ["panel", "calendar"]),
                            A("div", Fr, [
                              me(ee(Xe, {
                                months: I.value,
                                onUpdateMonth: T.value.next.setMonth
                              }, null, 8, ["months", "onUpdateMonth"]), [
                                [pe, d.next.month]
                              ]),
                              me(ee(qe, {
                                "as-prev-or-next": "",
                                years: T.value.next.years(),
                                onUpdateYear: T.value.next.setYear
                              }, null, 8, ["years", "onUpdateYear"]), [
                                [pe, d.next.year]
                              ]),
                              me(A("div", null, [
                                ee(Je, { weeks: b.value }, null, 8, ["weeks"]),
                                ee(Ze, {
                                  "as-prev-or-next": "",
                                  calendar: T.value.next,
                                  weeks: b.value,
                                  "as-range": j(),
                                  "week-number": p.weekNumber,
                                  onUpdateDate: (_) => E(_, u)
                                }, null, 8, ["calendar", "weeks", "as-range", "week-number", "onUpdateDate"])
                              ], 512), [
                                [pe, d.next.calendar]
                              ])
                            ])
                          ])) : ye("", !0)
                        ])
                      ]),
                      t.autoApply ? (z(), Z("div", Ur, [
                        A("div", zr, [
                          A("div", Wr, [
                            A("button", {
                              type: "button",
                              class: "away-cancel-picker w-full transition ease-out duration-300 inline-flex justify-center rounded-md border border-vtd-secondary-300 shadow-sm px-4 py-2 bg-white text-base font-medium text-vtd-secondary-700 hover:bg-vtd-secondary-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-vtd-primary-500 sm:mt-0 sm:ml-3 sm:w-auto sm:text-sm dark:ring-offset-vtd-secondary-800",
                              onClick: (_) => u(),
                              textContent: ae(t.options.footer.cancel)
                            }, null, 8, Kr)
                          ])
                        ])
                      ])) : (z(), Z("div", Br, [
                        A("div", Nr, [
                          A("div", Rr, [
                            A("button", {
                              type: "button",
                              class: "away-apply-picker w-full transition ease-out duration-300 inline-flex justify-center rounded-md border border-transparent shadow-sm px-4 py-2 bg-vtd-primary-600 text-base font-medium text-white hover:bg-vtd-primary-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-vtd-primary-500 sm:ml-3 sm:w-auto sm:text-sm dark:ring-offset-vtd-secondary-800 disabled:cursor-not-allowed",
                              disabled: t.asSingle ? y.value.length < 1 : y.value.length < 2,
                              onClick: (_) => R(u),
                              textContent: ae(t.options.footer.apply)
                            }, null, 8, Ir),
                            A("button", {
                              type: "button",
                              class: "mt-3 away-cancel-picker w-full transition ease-out duration-300 inline-flex justify-center rounded-md border border-vtd-secondary-300 shadow-sm px-4 py-2 bg-white text-base font-medium text-vtd-secondary-700 hover:bg-vtd-secondary-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-vtd-primary-500 sm:mt-0 sm:ml-3 sm:w-auto sm:text-sm dark:ring-offset-vtd-secondary-800",
                              onClick: (_) => u(),
                              textContent: ae(t.options.footer.cancel)
                            }, null, 8, Hr)
                          ])
                        ])
                      ]))
                    ], 2)
                  ], 512)
                ], 2)
              ]),
              _: 2
            }, 1024))
          ]),
          _: 2
        }, 1024)
      ]),
      _: 3
    }));
  }
});
const Gt = /* @__PURE__ */ (() => {
  const e = ua;
  return e.install = (r) => {
    r.component("VueTailwindDatepicker", e);
  }, e;
})(), da = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  default: Gt
}, Symbol.toStringTag, { value: "Module" }));
Object.entries(da).forEach(([e, r]) => {
  e !== "default" && (Gt[e] = r);
});
export {
  Gt as default
};
